
NGS I.
======

https://www.youtube.com/watch?v=9vNBYyHedLg
https://www.youtube.com/watch?v=ToKUGz\_YhC4
https://www.youtube.com/watch?v=9YxExTSwgPM

.. figure:: https://assets.illumina.com/content/dam/illumina-marketing/images/product/supporting-data-and-figures/nextera-mate-pair-library-prep-kit-web-graphic.png
   :alt: title

   title

Short Reads
-----------

SRA
~~~

https://www.ncbi.nlm.nih.gov/sra/docs/ példa: SRR1660257:
https://www.ncbi.nlm.nih.gov/sra/?term=SRR1660257

https://www.ncbi.nlm.nih.gov/geo/ https://www.ebi.ac.uk/ena

.. code:: bash

    %%bash 
    
    cd data/reads/
    #mv SRR1660257.fastq tt/
    fastq-dump SRR1660257


.. parsed-literal::

    Read 1015290 spots for SRR1660257
    Written 1015290 spots for SRR1660257


.. code:: bash

    %%bash 
    
    cd data
    mkdir reads
    cd reads
    
    #fastq-dump SRR1660257


.. parsed-literal::

    mkdir: cannot create directory ‘reads’: File exists
    bash: line 6: fastq-dump: command not found


.. code:: bash

    %%bash 
    
    cd data/reads
    
    ls -lh


.. parsed-literal::

    total 267M
    -rw-rw-r-- 1 sn sn 267M Mar 13 10:09 SRR1660257.fastq
    drwxrwxr-x 2 sn sn 4.0K Mar 13 10:08 tt


.. code:: bash

    %%bash 
    
    cd data/reads
    
    head SRR1660257.fastq


.. parsed-literal::

    @SRR1660257.1 1 length=100
    GTAATCTTAGATTCAAGTAATCCATATAATTAAATCCTGGTTTCATTGATTTGAGGCTGAGCTTTTCCATGAAGCAATCTGAAGAAAAACTAAAAGACAA
    +SRR1660257.1 1 length=100
    CBCFFFFFHHHHHJJIJHIJJJJJJJJJJJJJJJJJJJJJGIJJJJJJJJJJJIJJJJJIJJJJJJJJJJJJJJJIJJJJJJHHGHHHFFFFCEECDDDD
    @SRR1660257.2 2 length=100
    CAAGAGCACTGACTTCCTGGACCCCGCCACCACAACAAGTCCCCAAAACCACAGCGAGACCGCTGGCAACAACAACACTCATCACCAAGATACCGGAGAA
    +SRR1660257.2 2 length=100
    CCCFFFFFHHHHHJJJJJJJJJJJJJJJJJJJJIJJJJJIJJJJJJJIJJJJJJHHFFDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD
    @SRR1660257.3 3 length=100
    GCCCAAGATTGATCGAGGTTGGGTATGTGTTTTTCAGCTTCAAGATGGTAAAACACTTGGACTCAAAATTTGAGCCAATCTCCCTTCCCTCCGAAAGAGG


FASTQ
~~~~~

http://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp\_v2/Content/Vault/Informatics/Sequencing\_Analysis/BS/swSEQ\_mBS\_FASTQFiles.htm
https://en.wikipedia.org/wiki/FASTQ\_format

Minden readhez négy sor tartozik: - az első sorban van a read
azonosítója, illetve lehet még hozzá tartozó leírás - a második sorban
van a read nyers nukleotidsora - a harmadik + jellel kezdődő sor a jel
után lehet üres vagy a read azonosítóját is tartalmazhatja - a negyedik
sorban az egyes nukleotidok leolvasási minőségére vonatkozó kódolás
(részletek 74. lapon: Csaba Ortutay, Zsuzsanna Ortutay: Molecular Data
Analysis Using R. 2017, Wiley-Blackwell, ISBN: 978-1-119-16502-6.
https://www.wiley.com/en-hu/Molecular+Data+Analysis+Using+R-p-9781119165026)

BAM-ban tárolva:
https://gatkforums.broadinstitute.org/gatk/discussion/5990/what-is-ubam-and-why-is-it-better-than-fastq-for-storing-unmapped-sequence-data

CSFASTA
~~~~~~~

ABi SOLiD sequencer:
http://cutadapt.readthedocs.io/en/stable/colorspace.html

CSFASTA-file

::

    # Title: s0205_20110422_FRAG_BC_miRNA_MeDIP
    >1_5_224_F3
    T.222200232103..132..030.020..000.00
    >1_5_656_F3
    T.31311231.331..233..122.122..122.10
    >1_5_1005_F3"
    T.122221311011..212..312.033..303.32

+-----+-----+-----+-----+-----+
|     | A   | C   | G   | T   |
+=====+=====+=====+=====+=====+
| A   | 0   | 1   | 2   | 3   |
+-----+-----+-----+-----+-----+
| C   | 1   | 0   | 3   | 2   |
+-----+-----+-----+-----+-----+
| G   | 2   | 3   | 0   | 1   |
+-----+-----+-----+-----+-----+
| T   | 3   | 2   | 1   | 0   |
+-----+-----+-----+-----+-----+

QUAL-file

::

    # Title: s0205_20110422_FRAG_BC_miRNA_MeDIP
    >1_5_224_F3"
    -1 27 28 30 32 21 31 30 31 27 4 28 31 -1 -1 17 28 31 -1 -1 24 30 25 -1 22 25 21 -1 -1 30 28 21 -1 27 31
    >1_5_656_F3"
    -1 33 4 31 29 16 33 4 32 -1 20 33 4 -1 -1 25 31 10 -1 -1 12 25 5 -1 22 4 7 -1 -1 5 4 7 -1 4 4
    >1_5_1005_F3"
    -1 31 20 33 32 32 30 32 31 26 26 25 17 -1 -1 30 26 6 -1 -1 27 23 28 -1 25 31 10 -1 -1 29 14 19 -1 33 33

.. code:: bash

    %%bash 
    
    cd /home/sn/data/reads
    
    fastq-dump -X 10000 --split-files SRR1972739

.. code:: r

    library(ShortRead)
    library(seqTools)
    library(qrqc)
    library(BiocParallel)
    
    setwd('data/reads')
    
    fajlom = 'SRR1660257.fastq'
    
    readek = readFastq(fajlom)



.. parsed-literal::

    Loading required package: BiocGenerics
    Loading required package: parallel
    
    Attaching package: ‘BiocGenerics’
    
    The following objects are masked from ‘package:parallel’:
    
        clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
        clusterExport, clusterMap, parApply, parCapply, parLapply,
        parLapplyLB, parRapply, parSapply, parSapplyLB
    
    The following objects are masked from ‘package:stats’:
    
        IQR, mad, sd, var, xtabs
    
    The following objects are masked from ‘package:base’:
    
        anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,
        colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
        grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,
        mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
        rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
        table, tapply, union, unique, unsplit, which, which.max, which.min
    
    Loading required package: BiocParallel
    Loading required package: Biostrings
    Loading required package: S4Vectors
    Loading required package: stats4
    
    Attaching package: ‘S4Vectors’
    
    The following object is masked from ‘package:base’:
    
        expand.grid
    
    Loading required package: IRanges
    Loading required package: XVector
    
    Attaching package: ‘Biostrings’
    
    The following object is masked from ‘package:base’:
    
        strsplit
    
    Loading required package: Rsamtools
    Loading required package: GenomeInfoDb
    Loading required package: GenomicRanges
    Loading required package: GenomicAlignments
    Loading required package: SummarizedExperiment
    Loading required package: Biobase
    Welcome to Bioconductor
    
        Vignettes contain introductory material; view with
        'browseVignettes()'. To cite Bioconductor, see
        'citation("Biobase")', and for packages 'citation("pkgname")'.
    
    Loading required package: DelayedArray
    Loading required package: matrixStats
    
    Attaching package: ‘matrixStats’
    
    The following objects are masked from ‘package:Biobase’:
    
        anyMissing, rowMedians
    
    
    Attaching package: ‘DelayedArray’
    
    The following objects are masked from ‘package:matrixStats’:
    
        colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges
    
    The following object is masked from ‘package:Biostrings’:
    
        type
    
    The following object is masked from ‘package:base’:
    
        apply
    
    Loading required package: zlibbioc
    Loading required package: reshape
    
    Attaching package: ‘reshape’
    
    The following objects are masked from ‘package:S4Vectors’:
    
        expand, rename
    
    Loading required package: ggplot2
    Loading required package: biovizBase
    Loading required package: brew
    Loading required package: xtable
    Loading required package: testthat


.. code:: r

    slotNames(readek)



.. raw:: html

    <ol class=list-inline>
    	<li>'quality'</li>
    	<li>'sread'</li>
    	<li>'id'</li>
    </ol>



.. code:: r

    id(readek)



.. parsed-literal::

      A BStringSet instance of length 1015290
              width seq
          [1]    25 SRR1660257.1 1 length=100
          [2]    25 SRR1660257.2 2 length=100
          [3]    25 SRR1660257.3 3 length=100
          [4]    25 SRR1660257.4 4 length=100
          [5]    25 SRR1660257.5 5 length=100
          ...   ... ...
    [1015286]    37 SRR1660257.1015286 1015286 length=100
    [1015287]    37 SRR1660257.1015287 1015287 length=100
    [1015288]    37 SRR1660257.1015288 1015288 length=100
    [1015289]    37 SRR1660257.1015289 1015289 length=100
    [1015290]    37 SRR1660257.1015290 1015290 length=100


.. code:: r

    sread(readek)



.. parsed-literal::

      A DNAStringSet instance of length 1015290
              width seq
          [1]   100 GTAATCTTAGATTCAAGTAATCCATATAATT...GAAGCAATCTGAAGAAAAACTAAAAGACAA
          [2]   100 CAAGAGCACTGACTTCCTGGACCCCGCCACC...AACAACACTCATCACCAAGATACCGGAGAA
          [3]   100 GCCCAAGATTGATCGAGGTTGGGTATGTGTT...TGAGCCAATCTCCCTTCCCTCCGAAAGAGG
          [4]   100 TAATAATCAGATCTGCGAACCGGTAGAGTTT...CAATAGAAATTTAAACAGTGAGTGGAGACA
          [5]   100 CTTAGACATCAAAAATTCTTCCTGTTTTCGT...ATCCCATTGTTCCATGCTCATTCACTGATG
          ...   ... ...
    [1015286]   100 CAGATGATGAAGAGCAGGACAGGGACGGAAC...ATACAGAGATCACTCTGAAAAGAAAGAACT
    [1015287]   100 CTGGAAGTTCATAAGAATTTTCTTTTCCTGA...TATTGTTGGAGTTGCTTCTCAGCCTCAGTG
    [1015288]   100 ATCTTCCAAGATGCTGCTCCACCTGTCATCC...AAAGCTTGCGTCCAGTCCCACCATCGCCCA
    [1015289]   100 AGAGCCACAACTGAGCTACGCACCTTTTCAA...GGGGCGGCACATGCCACATTCTGGGACCGG
    [1015290]   100 GACGAGGACACTAAGCCGGTGCCTAATAGAT...GGCATATAGAGGGCAGACAGACACAATCCG


.. code:: r

    kval = quality(readek)
    kval



.. parsed-literal::

    class: FastqQuality
    quality:
      A BStringSet instance of length 1015290
              width seq
          [1]   100 CBCFFFFFHHHHHJJIJHIJJJJJJJJJJJJ...JJJJJIJJJJJJHHGHHHFFFFCEECDDDD
          [2]   100 CCCFFFFFHHHHHJJJJJJJJJJJJJJJJJJ...DDDDDDDDDDDDDDDDDDDDDDDDDDDDDD
          [3]   100 CCCFFFFFHHHHHJJIJJEGIJJCFGICFFG...HHFFFFECEEEEEDDDDDDDDDDDDDDDDB
          [4]   100 CCCFFFFFHHHHHJJJJGJJJJJGHIJJHIJ...EEEEEEDCCDDDDEDDDDCCDDACDDDDDD
          [5]   100 CCCFFFFFHHHHHJJJJJJJJJJJJJJJJIJ...JJGHHHHFFFFFFFECCCEEEEECCDDDDD
          ...   ... ...
    [1015286]   100 C@CFFFFFHHHHHJBHIIIJJIIIGIGGIIJ...DDDEEDDDCCDDDDDDDDDDDDDDDDDDDC
    [1015287]   100 ?@@FBDBBFHHBBGBECFFHIBEHEHDHHEG...CH>DF@@;;(..;AC@C>A@;C>9<ACC@:
    [1015288]   100 CCCFFFFFHHHHHJJJJJJJJIJJJHIIJJI...DEEDBCDDDDDDBDCCDDDDDDDCDDDDDD
    [1015289]   100 ;@@DDDDDHHFDCCGBHGI@GHFFCHHGIEC...A92=;59<@><@AACCCBAACDACCBBB<5
    [1015290]   100 1=@D4=A########################...##############################


PHRED minőségi pontszám:

:math:`Q_{PHRED} = -10 \times log_{10} P`

Ennek valószínűséggé való átalakítása:

:math:`P = 10^{-Q_{PHRED}/10}`

Néhány minőségi érték a döntésekhez:

+---------------------+---------+-------------+-----------------+
| :math:`Q_{PHRED}`   | hiba    |             | megbízhatóság   |
+=====================+=========+=============+=================+
| 10                  | 10%     | (1/10)      | 90%             |
+---------------------+---------+-------------+-----------------+
| 20                  | 1%      | (1/100)     | 99%             |
+---------------------+---------+-------------+-----------------+
| 30                  | 0.1%    | (1/1000)    | 99.9%           |
+---------------------+---------+-------------+-----------------+
| 40                  | 0.01%   | (1/10000)   | 99.99%          |
+---------------------+---------+-------------+-----------------+

Solexa:

:math:`Q_{Solexa}=-10\times log_{10}\left(\frac{P}{1-P}\right)`

Átváltások:

:math:`Q_{PHRED}=-10\times log_{10}\left(10^{Q_{Solexa}/10}+1\right)`

:math:`Q_{Solexa}=-10\times log_{10}\left(10^{Q_{PHRED}/10}-1\right)`

.. raw:: html

   <table style="width:60%">

.. raw:: html

   <tr>

::

    <th></th>
    <th>ASCII</th> 
    <th>Minőségi pont</th>    
    <th></th>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <th></th>
    <th>Tartomány</th> 
    <th>Típus</th>    
    <th>Tartomány</th>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>Sanger standard</td>
    <td></td>
    <td></td>
    <td></td>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>  fastq-sanger</td>
    <td>33 - 126</td>
    <td>PHRED</td>
    <td>0 - 93</td>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>Solexa/régebbi Illumina</td>
    <td></td>
    <td></td>
    <td></td>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>  fastq-solexa</td>
    <td>59 - 126</td>
    <td>Solexa</td>
    <td>-5 - 62</td>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>Illumina 1.3+</td>
    <td></td>
    <td></td>
    <td></td>

.. raw:: html

   </tr>

.. raw:: html

   <tr>

::

    <td>  fastq-illumina</td>
    <td>64 - 126</td>
    <td>PHRED</td>
    <td>0 - 62</td>

.. raw:: html

   </tr>

.. raw:: html

   </table>

Cock PJA, Fields CJ, Goto N, Heuer ML, Rice PM The sanger FASTQ file
format for sequences with quality scores and the Solexa/Illumina FASTQ
variants. Nucleic Acids Research 2010 Vol.38 No.6 1767-1771

.. code:: r

    phredTable()



.. raw:: html

    <table>
    <thead><tr><th scope=col>ascii</th><th scope=col>phred</th><th scope=col>char</th></tr></thead>
    <tbody>
    	<tr><td>33</td><td> 0</td><td>! </td></tr>
    	<tr><td>34</td><td> 1</td><td>" </td></tr>
    	<tr><td>35</td><td> 2</td><td># </td></tr>
    	<tr><td>36</td><td> 3</td><td>$ </td></tr>
    	<tr><td>37</td><td> 4</td><td>% </td></tr>
    	<tr><td>38   </td><td> 5   </td><td>&amp;</td></tr>
    	<tr><td>39</td><td> 6</td><td>' </td></tr>
    	<tr><td>40</td><td> 7</td><td>( </td></tr>
    	<tr><td>41</td><td> 8</td><td>) </td></tr>
    	<tr><td>42</td><td> 9</td><td>* </td></tr>
    	<tr><td>43</td><td>10</td><td>+ </td></tr>
    	<tr><td>44</td><td>11</td><td>, </td></tr>
    	<tr><td>45</td><td>12</td><td>- </td></tr>
    	<tr><td>46</td><td>13</td><td>. </td></tr>
    	<tr><td>47</td><td>14</td><td>/ </td></tr>
    	<tr><td>48</td><td>15</td><td>0 </td></tr>
    	<tr><td>49</td><td>16</td><td>1 </td></tr>
    	<tr><td>50</td><td>17</td><td>2 </td></tr>
    	<tr><td>51</td><td>18</td><td>3 </td></tr>
    	<tr><td>52</td><td>19</td><td>4 </td></tr>
    	<tr><td>53</td><td>20</td><td>5 </td></tr>
    	<tr><td>54</td><td>21</td><td>6 </td></tr>
    	<tr><td>55</td><td>22</td><td>7 </td></tr>
    	<tr><td>56</td><td>23</td><td>8 </td></tr>
    	<tr><td>57</td><td>24</td><td>9 </td></tr>
    	<tr><td>58</td><td>25</td><td>: </td></tr>
    	<tr><td>59</td><td>26</td><td>; </td></tr>
    	<tr><td>60  </td><td>27  </td><td>&lt;</td></tr>
    	<tr><td>61</td><td>28</td><td>= </td></tr>
    	<tr><td>62  </td><td>29  </td><td>&gt;</td></tr>
    	<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>
    	<tr><td> 97</td><td>64 </td><td>a  </td></tr>
    	<tr><td> 98</td><td>65 </td><td>b  </td></tr>
    	<tr><td> 99</td><td>66 </td><td>c  </td></tr>
    	<tr><td>100</td><td>67 </td><td>d  </td></tr>
    	<tr><td>101</td><td>68 </td><td>e  </td></tr>
    	<tr><td>102</td><td>69 </td><td>f  </td></tr>
    	<tr><td>103</td><td>70 </td><td>g  </td></tr>
    	<tr><td>104</td><td>71 </td><td>h  </td></tr>
    	<tr><td>105</td><td>72 </td><td>i  </td></tr>
    	<tr><td>106</td><td>73 </td><td>j  </td></tr>
    	<tr><td>107</td><td>74 </td><td>k  </td></tr>
    	<tr><td>108</td><td>75 </td><td>l  </td></tr>
    	<tr><td>109</td><td>76 </td><td>m  </td></tr>
    	<tr><td>110</td><td>77 </td><td>n  </td></tr>
    	<tr><td>111</td><td>78 </td><td>o  </td></tr>
    	<tr><td>112</td><td>79 </td><td>p  </td></tr>
    	<tr><td>113</td><td>80 </td><td>q  </td></tr>
    	<tr><td>114</td><td>81 </td><td>r  </td></tr>
    	<tr><td>115</td><td>82 </td><td>s  </td></tr>
    	<tr><td>116</td><td>83 </td><td>t  </td></tr>
    	<tr><td>117</td><td>84 </td><td>u  </td></tr>
    	<tr><td>118</td><td>85 </td><td>v  </td></tr>
    	<tr><td>119</td><td>86 </td><td>w  </td></tr>
    	<tr><td>120</td><td>87 </td><td>x  </td></tr>
    	<tr><td>121</td><td>88 </td><td>y  </td></tr>
    	<tr><td>122</td><td>89 </td><td>z  </td></tr>
    	<tr><td>123</td><td>90 </td><td>{  </td></tr>
    	<tr><td>124</td><td>91 </td><td>|  </td></tr>
    	<tr><td>125</td><td>92 </td><td>}  </td></tr>
    	<tr><td>126</td><td>93 </td><td>~  </td></tr>
    </tbody>
    </table>



.. code:: r

    m = as(kval, 'matrix')
    m[1,]



.. raw:: html

    <ol class=list-inline>
    	<li>34</li>
    	<li>33</li>
    	<li>34</li>
    	<li>37</li>
    	<li>37</li>
    	<li>37</li>
    	<li>37</li>
    	<li>37</li>
    	<li>39</li>
    	<li>39</li>
    	<li>39</li>
    	<li>39</li>
    	<li>39</li>
    	<li>41</li>
    	<li>41</li>
    	<li>40</li>
    	<li>41</li>
    	<li>39</li>
    	<li>40</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>38</li>
    	<li>40</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>40</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>40</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>40</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>41</li>
    	<li>39</li>
    	<li>39</li>
    	<li>38</li>
    	<li>39</li>
    	<li>39</li>
    	<li>39</li>
    	<li>37</li>
    	<li>37</li>
    	<li>37</li>
    	<li>37</li>
    	<li>34</li>
    	<li>36</li>
    	<li>36</li>
    	<li>34</li>
    	<li>35</li>
    	<li>35</li>
    	<li>35</li>
    	<li>35</li>
    </ol>


