
NGS II.
=======

Readek minőségellenőrzése
^^^^^^^^^^^^^^^^^^^^^^^^^

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2752612/

FastQC: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
https://www.youtube.com/watch?v=bz93ReOv87Y

.. code:: bash

    %%bash
    
    mkdir gyak06

.. code:: bash

    %%bash
    
    cd gyak06
    
    export PATH=$PATH:/usr/local/ncbi/sra-tools/bin
    
    fastq-dump SRR1660259
    
    ls -lh


.. parsed-literal::

    Read 9130 spots for SRR1660259
    Written 9130 spots for SRR1660259
    total 2.4M
    -rw-rw-r-- 1 sn sn 2.4M Mar 20 09:31 SRR1660259.fastq


.. code:: bash

    %%bash
    
    cd gyak06
    
    ls -lh



.. parsed-literal::

    total 2.4M
    -rw-rw-r-- 1 sn sn 2.4M Mar 20 09:31 SRR1660259.fastq


.. code:: ipython3

    # R
    setwd('gyak06')
    
    library(fastqcr)
    
    fastqc() 

.. code:: ipython3

    # R
    minosegi.fajl = 'FASTQC/SRR1660259_fastqc.zip'
    
    qc_report(
        qc.path = minosegi.fajl, 
        template = 'minta_jelentes.Rmd', 
        result.file ='osszefoglalo', 
        interpret = TRUE, 
        experiment = 'minőségellenőrzési példa',
        preview = FALSE
    )


.. parsed-literal::

    
    
    processing file: minta_jelentes.Rmd


.. parsed-literal::

      |.                                                                |   2%
       inline R code fragments
    
      |..                                                               |   3%
    label: unnamed-chunk-1 (with options) 
    List of 1
     $ echo: logi FALSE
    
      |...                                                              |   5%
      ordinary text without R code
    
      |....                                                             |   6%
    label: unnamed-chunk-2 (with options) 
    List of 1
     $ echo: logi FALSE
    
      |.....                                                            |   8%
       inline R code fragments
    
      |......                                                           |  10%
    label: unnamed-chunk-3 (with options) 
    List of 1
     $ echo: logi TRUE
    


.. parsed-literal::

    
    Attaching package: 'dplyr'
    
    The following objects are masked from 'package:stats':
    
        filter, lag
    
    The following objects are masked from 'package:base':
    
        intersect, setdiff, setequal, union
    


.. parsed-literal::

      |.......                                                          |  11%
      ordinary text without R code
    
      |........                                                         |  13%
    label: unnamed-chunk-4


.. parsed-literal::

    Reading: FASTQC/SRR1660259_fastqc.zip


.. parsed-literal::

      |.........                                                        |  14%
      ordinary text without R code
    
      |..........                                                       |  16%
    label: summary (with options) 
    List of 3
     $ fig.width : num 5
     $ fig.height: num 4.5
     $ fig.align : chr "center"
    
      |...........                                                      |  17%
      ordinary text without R code
    
      |............                                                     |  19%
    label: basic-statistics (with options) 
    List of 3
     $ fig.width : num 5
     $ fig.height: num 3
     $ fig.align : chr "center"
    
      |.............                                                    |  21%
      ordinary text without R code
    
      |..............                                                   |  22%
    label: per-base-sequence-quality (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |...............                                                  |  24%
      ordinary text without R code
    
      |.................                                                |  25%
    label: unnamed-chunk-5 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |..................                                               |  27%
      ordinary text without R code
    
      |...................                                              |  29%
    label: unnamed-chunk-6 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |....................                                             |  30%
      ordinary text without R code
    
      |.....................                                            |  32%
    label: per-sequence-quality-scores (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |......................                                           |  33%
      ordinary text without R code
    
      |.......................                                          |  35%
    label: unnamed-chunk-7 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |........................                                         |  37%
      ordinary text without R code
    
      |.........................                                        |  38%
    label: unnamed-chunk-8 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |..........................                                       |  40%
      ordinary text without R code
    
      |...........................                                      |  41%
    label: per-base-sequence-content (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |............................                                     |  43%
      ordinary text without R code
    
      |.............................                                    |  44%
    label: unnamed-chunk-9 (with options) 
    List of 2
     $ type  : chr "notice"
     $ engine: chr "block"
    
      |..............................                                   |  46%
      ordinary text without R code
    
      |...............................                                  |  48%
    label: unnamed-chunk-10 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |................................                                 |  49%
      ordinary text without R code
    
      |.................................                                |  51%
    label: unnamed-chunk-11 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |..................................                               |  52%
      ordinary text without R code
    
      |...................................                              |  54%
    label: per-sequence-GC-content (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |....................................                             |  56%
      ordinary text without R code
    
      |.....................................                            |  57%
    label: unnamed-chunk-12 (with options) 
    List of 2
     $ type  : chr "success"
     $ engine: chr "block"
    
      |......................................                           |  59%
      ordinary text without R code
    
      |.......................................                          |  60%
    label: per-base-N-content (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |........................................                         |  62%
      ordinary text without R code
    
      |.........................................                        |  63%
    label: unnamed-chunk-13 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |..........................................                       |  65%
      ordinary text without R code
    
      |...........................................                      |  67%
    label: unnamed-chunk-14 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |............................................                     |  68%
      ordinary text without R code
    
      |.............................................                    |  70%
    label: sequence-length-distribution (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    


.. parsed-literal::

    geom_path: Each group consists of only one observation. Do you need to adjust the group aesthetic?


.. parsed-literal::

      |..............................................                   |  71%
      ordinary text without R code
    
      |...............................................                  |  73%
    label: sequence-duplication-levels (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |................................................                 |  75%
      ordinary text without R code
    
      |..................................................               |  76%
    label: unnamed-chunk-15 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |...................................................              |  78%
      ordinary text without R code
    
      |....................................................             |  79%
    label: unnamed-chunk-16 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |.....................................................            |  81%
      ordinary text without R code
    
      |......................................................           |  83%
    label: Overrepresented-sequences (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |.......................................................          |  84%
      ordinary text without R code
    
      |........................................................         |  86%
    label: unnamed-chunk-17 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |.........................................................        |  87%
      ordinary text without R code
    
      |..........................................................       |  89%
    label: unnamed-chunk-18 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |...........................................................      |  90%
      ordinary text without R code
    
      |............................................................     |  92%
    label: adapter-content (with options) 
    List of 3
     $ fig.width : num 4
     $ fig.height: num 3.5
     $ fig.align : chr "center"
    
      |.............................................................    |  94%
      ordinary text without R code
    
      |..............................................................   |  95%
    label: unnamed-chunk-19 (with options) 
    List of 2
     $ type  : chr "warning"
     $ engine: chr "block"
    
      |...............................................................  |  97%
      ordinary text without R code
    
      |................................................................ |  98%
    label: unnamed-chunk-20 (with options) 
    List of 2
     $ type  : chr "block"
     $ engine: chr "block"
    
      |.................................................................| 100%
      ordinary text without R code
    
    


.. parsed-literal::

    output file: minta_jelentes.knit.md
    


.. parsed-literal::

    /usr/bin/pandoc +RTS -K512m -RTS minta_jelentes.utf8.md --to html4 --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output /home/sn/gyak06/osszefoglalo.html --smart --email-obfuscation none --self-contained --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /usr/local/lib/R/library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --variable 'theme:bootstrap' --include-in-header /tmp/RtmpLf2JIY/rmarkdown-str86a7a75972b.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' 


.. parsed-literal::

    
    Output created: osszefoglalo.html
    
    --------------------------
    Output file: /home/sn/gyak06/osszefoglalo.html
    --------------------------
    


.. code:: ipython3

    qc = qc_read(minosegi.fajl)
    qc


.. parsed-literal::

    Reading: FASTQC/SRR1660259_fastqc.zip



.. raw:: html

    <dl>
    	<dt>$summary</dt>
    		<dd><table>
    <thead><tr><th scope=col>status</th><th scope=col>module</th><th scope=col>sample</th></tr></thead>
    <tbody>
    	<tr><td>PASS                        </td><td>Basic Statistics            </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>PASS                        </td><td>Per base sequence quality   </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>PASS                        </td><td>Per sequence quality scores </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>WARN                        </td><td>Per base sequence content   </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>FAIL                        </td><td>Per sequence GC content     </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>PASS                        </td><td>Per base N content          </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>WARN                        </td><td>Sequence Length Distribution</td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>WARN                        </td><td>Sequence Duplication Levels </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>FAIL                        </td><td>Overrepresented sequences   </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>PASS                        </td><td>Adapter Content             </td><td>SRR1660259.fastq            </td></tr>
    	<tr><td>WARN                        </td><td>Kmer Content                </td><td>SRR1660259.fastq            </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$basic_statistics</dt>
    		<dd><table>
    <thead><tr><th scope=col>Measure</th><th scope=col>Value</th></tr></thead>
    <tbody>
    	<tr><td>Filename                         </td><td>SRR1660259.fastq                 </td></tr>
    	<tr><td>File type                        </td><td>Conventional base calls          </td></tr>
    	<tr><td>Encoding                         </td><td>Sanger / Illumina 1.9            </td></tr>
    	<tr><td>Total Sequences                  </td><td>9130                             </td></tr>
    	<tr><td>Sequences flagged as poor quality</td><td>0                                </td></tr>
    	<tr><td>Sequence length                  </td><td>49-100                           </td></tr>
    	<tr><td>%GC                              </td><td>42                               </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$per_base_sequence_quality</dt>
    		<dd><table>
    <thead><tr><th scope=col>Base</th><th scope=col>Mean</th><th scope=col>Median</th><th scope=col>Lower Quartile</th><th scope=col>Upper Quartile</th><th scope=col>10th Percentile</th><th scope=col>90th Percentile</th></tr></thead>
    <tbody>
    	<tr><td>1       </td><td>32.66123</td><td>34.0    </td><td>31.0    </td><td>34.0    </td><td>31.0    </td><td>34.0    </td></tr>
    	<tr><td>2       </td><td>31.48335</td><td>34.0    </td><td>31.0    </td><td>34.0    </td><td>28.0    </td><td>34.0    </td></tr>
    	<tr><td>3       </td><td>31.87163</td><td>34.0    </td><td>31.0    </td><td>34.0    </td><td>28.0    </td><td>34.0    </td></tr>
    	<tr><td>4       </td><td>36.08510</td><td>37.0    </td><td>37.0    </td><td>37.0    </td><td>33.0    </td><td>37.0    </td></tr>
    	<tr><td>5       </td><td>36.24940</td><td>37.0    </td><td>37.0    </td><td>37.0    </td><td>35.0    </td><td>37.0    </td></tr>
    	<tr><td>6       </td><td>36.29803</td><td>37.0    </td><td>37.0    </td><td>37.0    </td><td>35.0    </td><td>37.0    </td></tr>
    	<tr><td>7       </td><td>36.39025</td><td>37.0    </td><td>37.0    </td><td>37.0    </td><td>35.0    </td><td>37.0    </td></tr>
    	<tr><td>8       </td><td>36.38576</td><td>37.0    </td><td>37.0    </td><td>37.0    </td><td>35.0    </td><td>37.0    </td></tr>
    	<tr><td>9       </td><td>38.16276</td><td>39.0    </td><td>39.0    </td><td>39.0    </td><td>37.0    </td><td>39.0    </td></tr>
    	<tr><td>10-11   </td><td>38.21002</td><td>39.0    </td><td>39.0    </td><td>39.0    </td><td>37.0    </td><td>39.0    </td></tr>
    	<tr><td>12-13   </td><td>38.19655</td><td>39.0    </td><td>39.0    </td><td>39.0    </td><td>37.0    </td><td>39.0    </td></tr>
    	<tr><td>14-15   </td><td>39.81402</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>38.0    </td><td>41.0    </td></tr>
    	<tr><td>16-17   </td><td>39.80767</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>38.0    </td><td>41.0    </td></tr>
    	<tr><td>18-19   </td><td>39.71840</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>20-21   </td><td>39.69496</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>22-23   </td><td>39.71846</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>24-25   </td><td>39.68647</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>26-27   </td><td>39.55586</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>28-29   </td><td>39.52152</td><td>41.0    </td><td>40.0    </td><td>41.0    </td><td>37.0    </td><td>41.0    </td></tr>
    	<tr><td>30-31   </td><td>39.39250</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>36.5    </td><td>41.0    </td></tr>
    	<tr><td>32-33   </td><td>39.38532</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>36.0    </td><td>41.0    </td></tr>
    	<tr><td>34-35   </td><td>39.28122</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>36.0    </td><td>41.0    </td></tr>
    	<tr><td>36-37   </td><td>39.17503</td><td>40.0    </td><td>39.0    </td><td>41.0    </td><td>36.0    </td><td>41.0    </td></tr>
    	<tr><td>38-39   </td><td>39.21172</td><td>40.5    </td><td>39.0    </td><td>41.0    </td><td>36.0    </td><td>41.0    </td></tr>
    	<tr><td>40-41   </td><td>39.34036</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>36.0    </td><td>41.0    </td></tr>
    	<tr><td>42-43   </td><td>39.17651</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>35.5    </td><td>41.0    </td></tr>
    	<tr><td>44-45   </td><td>39.11873</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>46-47   </td><td>39.06802</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>48-49   </td><td>39.03297</td><td>41.0    </td><td>39.0    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>50-51   </td><td>38.91149</td><td>40.5    </td><td>38.5    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>52-53   </td><td>38.77225</td><td>40.0    </td><td>38.0    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>54-55   </td><td>38.49502</td><td>40.0    </td><td>38.0    </td><td>41.0    </td><td>34.0    </td><td>41.0    </td></tr>
    	<tr><td>56-57   </td><td>38.54760</td><td>40.0    </td><td>38.0    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>58-59   </td><td>38.48814</td><td>40.0    </td><td>37.5    </td><td>41.0    </td><td>35.0    </td><td>41.0    </td></tr>
    	<tr><td>60-61   </td><td>38.28277</td><td>40.0    </td><td>37.0    </td><td>41.0    </td><td>34.0    </td><td>41.0    </td></tr>
    	<tr><td>62-63   </td><td>37.99331</td><td>39.5    </td><td>36.5    </td><td>41.0    </td><td>34.0    </td><td>41.0    </td></tr>
    	<tr><td>64-65   </td><td>37.65415</td><td>39.0    </td><td>36.0    </td><td>41.0    </td><td>34.0    </td><td>41.0    </td></tr>
    	<tr><td>66-67   </td><td>37.33899</td><td>39.0    </td><td>35.0    </td><td>41.0    </td><td>34.0    </td><td>41.0    </td></tr>
    	<tr><td>68-69   </td><td>36.97357</td><td>37.5    </td><td>35.0    </td><td>40.0    </td><td>33.5    </td><td>41.0    </td></tr>
    	<tr><td>70-71   </td><td>36.55212</td><td>37.0    </td><td>35.0    </td><td>39.5    </td><td>33.0    </td><td>41.0    </td></tr>
    	<tr><td>72-73   </td><td>36.20034</td><td>37.0    </td><td>35.0    </td><td>39.0    </td><td>33.0    </td><td>41.0    </td></tr>
    	<tr><td>74-75   </td><td>35.80002</td><td>36.0    </td><td>35.0    </td><td>39.0    </td><td>33.0    </td><td>40.5    </td></tr>
    	<tr><td>76-77   </td><td>34.90666</td><td>35.0    </td><td>34.5    </td><td>37.0    </td><td>31.0    </td><td>39.0    </td></tr>
    	<tr><td>78-79   </td><td>34.70149</td><td>35.0    </td><td>35.0    </td><td>37.0    </td><td>31.0    </td><td>39.0    </td></tr>
    	<tr><td>80-81   </td><td>34.25147</td><td>35.0    </td><td>34.5    </td><td>36.5    </td><td>30.5    </td><td>38.0    </td></tr>
    	<tr><td>82-83   </td><td>33.73188</td><td>35.0    </td><td>34.0    </td><td>36.0    </td><td>29.5    </td><td>37.0    </td></tr>
    	<tr><td>84-85   </td><td>33.59272</td><td>35.0    </td><td>34.0    </td><td>36.0    </td><td>30.0    </td><td>37.0    </td></tr>
    	<tr><td>86-87   </td><td>33.56807</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>30.5    </td><td>36.0    </td></tr>
    	<tr><td>88-89   </td><td>33.34040</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>30.0    </td><td>36.0    </td></tr>
    	<tr><td>90-91   </td><td>33.09901</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>30.0    </td><td>36.0    </td></tr>
    	<tr><td>92-93   </td><td>32.78568</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>29.0    </td><td>36.0    </td></tr>
    	<tr><td>94-95   </td><td>32.81963</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>30.0    </td><td>35.5    </td></tr>
    	<tr><td>96-97   </td><td>32.43062</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>29.0    </td><td>35.0    </td></tr>
    	<tr><td>98-99   </td><td>32.29074</td><td>35.0    </td><td>34.0    </td><td>35.0    </td><td>29.0    </td><td>35.0    </td></tr>
    	<tr><td>100     </td><td>30.90465</td><td>34.0    </td><td>31.0    </td><td>35.0    </td><td>24.0    </td><td>35.0    </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$per_tile_sequence_quality</dt>
    		<dd></dd>
    	<dt>$per_sequence_quality_scores</dt>
    		<dd><table>
    <thead><tr><th scope=col>Quality</th><th scope=col>Count</th></tr></thead>
    <tbody>
    	<tr><td> 8  </td><td>   1</td></tr>
    	<tr><td> 9  </td><td>   1</td></tr>
    	<tr><td>10  </td><td>   0</td></tr>
    	<tr><td>11  </td><td>   1</td></tr>
    	<tr><td>12  </td><td>   2</td></tr>
    	<tr><td>13  </td><td>   3</td></tr>
    	<tr><td>14  </td><td>   5</td></tr>
    	<tr><td>15  </td><td>   5</td></tr>
    	<tr><td>16  </td><td>   4</td></tr>
    	<tr><td>17  </td><td>   6</td></tr>
    	<tr><td>18  </td><td>   8</td></tr>
    	<tr><td>19  </td><td>  11</td></tr>
    	<tr><td>20  </td><td>   7</td></tr>
    	<tr><td>21  </td><td>  11</td></tr>
    	<tr><td>22  </td><td>  13</td></tr>
    	<tr><td>23  </td><td>  15</td></tr>
    	<tr><td>24  </td><td>  18</td></tr>
    	<tr><td>25  </td><td>  25</td></tr>
    	<tr><td>26  </td><td>  50</td></tr>
    	<tr><td>27  </td><td>  49</td></tr>
    	<tr><td>28  </td><td>  47</td></tr>
    	<tr><td>29  </td><td>  82</td></tr>
    	<tr><td>30  </td><td> 103</td></tr>
    	<tr><td>31  </td><td> 121</td></tr>
    	<tr><td>32  </td><td> 134</td></tr>
    	<tr><td>33  </td><td> 214</td></tr>
    	<tr><td>34  </td><td> 288</td></tr>
    	<tr><td>35  </td><td> 432</td></tr>
    	<tr><td>36  </td><td> 842</td></tr>
    	<tr><td>37  </td><td>1991</td></tr>
    	<tr><td>38  </td><td>3717</td></tr>
    	<tr><td>39  </td><td> 922</td></tr>
    	<tr><td>40  </td><td>   2</td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$per_base_sequence_content</dt>
    		<dd><table>
    <thead><tr><th scope=col>Base</th><th scope=col>G</th><th scope=col>A</th><th scope=col>T</th><th scope=col>C</th></tr></thead>
    <tbody>
    	<tr><td>1       </td><td>28.46659</td><td>16.45126</td><td>19.79189</td><td>35.29025</td></tr>
    	<tr><td>2       </td><td>26.76889</td><td>20.70099</td><td>32.94633</td><td>19.58379</td></tr>
    	<tr><td>3       </td><td>22.27820</td><td>26.41840</td><td>27.89704</td><td>23.40635</td></tr>
    	<tr><td>4       </td><td>24.55641</td><td>26.30887</td><td>23.91019</td><td>25.22453</td></tr>
    	<tr><td>5       </td><td>23.81161</td><td>30.64622</td><td>25.69551</td><td>19.84666</td></tr>
    	<tr><td>6       </td><td>19.29901</td><td>35.43264</td><td>26.69222</td><td>18.57612</td></tr>
    	<tr><td>7       </td><td>18.59803</td><td>23.27492</td><td>40.21906</td><td>17.90800</td></tr>
    	<tr><td>8       </td><td>17.20701</td><td>27.88609</td><td>32.08105</td><td>22.82585</td></tr>
    	<tr><td>9       </td><td>21.89485</td><td>25.89266</td><td>30.84337</td><td>21.36911</td></tr>
    	<tr><td>10-11   </td><td>24.12377</td><td>32.69989</td><td>23.72399</td><td>19.45235</td></tr>
    	<tr><td>12-13   </td><td>19.91785</td><td>28.48850</td><td>30.48740</td><td>21.10624</td></tr>
    	<tr><td>14-15   </td><td>20.94743</td><td>28.68565</td><td>30.56955</td><td>19.79737</td></tr>
    	<tr><td>16-17   </td><td>23.03395</td><td>27.36583</td><td>29.43045</td><td>20.16977</td></tr>
    	<tr><td>18-19   </td><td>22.09748</td><td>27.22344</td><td>29.28806</td><td>21.39102</td></tr>
    	<tr><td>20-21   </td><td>22.65608</td><td>28.71851</td><td>28.23658</td><td>20.38883</td></tr>
    	<tr><td>22-23   </td><td>20.08762</td><td>29.90142</td><td>29.84666</td><td>20.16429</td></tr>
    	<tr><td>24-25   </td><td>21.97700</td><td>27.42059</td><td>28.85542</td><td>21.74699</td></tr>
    	<tr><td>26-27   </td><td>24.57831</td><td>28.88828</td><td>27.68346</td><td>18.84995</td></tr>
    	<tr><td>28-29   </td><td>20.71742</td><td>30.88719</td><td>28.12705</td><td>20.26835</td></tr>
    	<tr><td>30-31   </td><td>21.81271</td><td>30.85433</td><td>27.64513</td><td>19.68784</td></tr>
    	<tr><td>32-33   </td><td>21.12815</td><td>30.99124</td><td>28.66922</td><td>19.21139</td></tr>
    	<tr><td>34-35   </td><td>24.43045</td><td>28.63636</td><td>27.56298</td><td>19.37021</td></tr>
    	<tr><td>36-37   </td><td>21.11720</td><td>29.47974</td><td>29.70427</td><td>19.69880</td></tr>
    	<tr><td>38-39   </td><td>22.74918</td><td>27.58488</td><td>31.09529</td><td>18.57065</td></tr>
    	<tr><td>40-41   </td><td>21.19934</td><td>29.34830</td><td>30.75575</td><td>18.69660</td></tr>
    	<tr><td>42-43   </td><td>21.99343</td><td>28.48302</td><td>29.30449</td><td>20.21906</td></tr>
    	<tr><td>44-45   </td><td>22.69441</td><td>29.28258</td><td>28.42278</td><td>19.60022</td></tr>
    	<tr><td>46-47   </td><td>22.93538</td><td>29.27163</td><td>28.74589</td><td>19.04710</td></tr>
    	<tr><td>48-49   </td><td>18.70756</td><td>30.62979</td><td>28.55969</td><td>22.10296</td></tr>
    	<tr><td>50-51   </td><td>21.61244</td><td>28.46423</td><td>30.17855</td><td>19.74477</td></tr>
    	<tr><td>52-53   </td><td>21.84915</td><td>30.49241</td><td>27.75374</td><td>19.90469</td></tr>
    	<tr><td>54-55   </td><td>20.10298</td><td>28.03462</td><td>31.16236</td><td>20.70004</td></tr>
    	<tr><td>56-57   </td><td>19.46209</td><td>28.70837</td><td>31.48006</td><td>20.34947</td></tr>
    	<tr><td>58-59   </td><td>20.42728</td><td>28.54013</td><td>28.98384</td><td>22.04875</td></tr>
    	<tr><td>60-61   </td><td>20.08657</td><td>29.25867</td><td>28.69432</td><td>21.96044</td></tr>
    	<tr><td>62-63   </td><td>19.19114</td><td>31.17602</td><td>28.14555</td><td>21.48729</td></tr>
    	<tr><td>64-65   </td><td>19.83779</td><td>29.82793</td><td>28.90180</td><td>21.43249</td></tr>
    	<tr><td>66-67   </td><td>21.80871</td><td>28.12825</td><td>28.44067</td><td>21.62236</td></tr>
    	<tr><td>68-69   </td><td>20.38925</td><td>30.42763</td><td>28.93092</td><td>20.25219</td></tr>
    	<tr><td>70-71   </td><td>20.53194</td><td>29.10337</td><td>27.99013</td><td>22.37455</td></tr>
    	<tr><td>72-73   </td><td>20.26434</td><td>26.73577</td><td>29.00077</td><td>23.99912</td></tr>
    	<tr><td>74-75   </td><td>21.13317</td><td>30.27095</td><td>28.62549</td><td>19.97038</td></tr>
    	<tr><td>76-77   </td><td>20.87357</td><td>28.12774</td><td>29.92208</td><td>21.07660</td></tr>
    	<tr><td>78-79   </td><td>23.44162</td><td>27.17296</td><td>26.77239</td><td>22.61304</td></tr>
    	<tr><td>80-81   </td><td>20.79605</td><td>29.38238</td><td>28.53692</td><td>21.28466</td></tr>
    	<tr><td>82-83   </td><td>21.65532</td><td>29.15202</td><td>28.55338</td><td>20.63928</td></tr>
    	<tr><td>84-85   </td><td>19.50657</td><td>28.31474</td><td>30.52915</td><td>21.64954</td></tr>
    	<tr><td>86-87   </td><td>20.80172</td><td>29.35775</td><td>28.40097</td><td>21.43957</td></tr>
    	<tr><td>88-89   </td><td>20.62933</td><td>30.15733</td><td>26.91165</td><td>22.30168</td></tr>
    	<tr><td>90-91   </td><td>21.64023</td><td>28.14497</td><td>28.62415</td><td>21.59066</td></tr>
    	<tr><td>92-93   </td><td>20.13349</td><td>29.51624</td><td>27.65734</td><td>22.69292</td></tr>
    	<tr><td>94-95   </td><td>22.41741</td><td>28.98575</td><td>27.20141</td><td>21.39543</td></tr>
    	<tr><td>96-97   </td><td>23.98078</td><td>27.17932</td><td>26.88653</td><td>21.95338</td></tr>
    	<tr><td>98-99   </td><td>20.57231</td><td>32.14010</td><td>25.51652</td><td>21.77108</td></tr>
    	<tr><td>100     </td><td>21.08054</td><td>27.40029</td><td>30.21766</td><td>21.30151</td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$per_sequence_gc_content</dt>
    		<dd><table>
    <thead><tr><th scope=col>GC Content</th><th scope=col>Count</th></tr></thead>
    <tbody>
    	<tr><td> 0  </td><td> 0.0</td></tr>
    	<tr><td> 1  </td><td> 0.0</td></tr>
    	<tr><td> 2  </td><td> 0.0</td></tr>
    	<tr><td> 3  </td><td> 0.0</td></tr>
    	<tr><td> 4  </td><td> 0.0</td></tr>
    	<tr><td> 5  </td><td> 0.0</td></tr>
    	<tr><td> 6  </td><td> 0.0</td></tr>
    	<tr><td> 7  </td><td> 0.0</td></tr>
    	<tr><td> 8  </td><td> 0.0</td></tr>
    	<tr><td> 9  </td><td> 0.0</td></tr>
    	<tr><td>10  </td><td> 0.0</td></tr>
    	<tr><td>11  </td><td> 0.0</td></tr>
    	<tr><td>12  </td><td> 0.0</td></tr>
    	<tr><td>13  </td><td> 0.0</td></tr>
    	<tr><td>14  </td><td> 0.0</td></tr>
    	<tr><td>15  </td><td> 0.0</td></tr>
    	<tr><td>16  </td><td> 0.0</td></tr>
    	<tr><td>17  </td><td> 0.0</td></tr>
    	<tr><td>18  </td><td> 0.0</td></tr>
    	<tr><td>19  </td><td> 0.0</td></tr>
    	<tr><td>20  </td><td> 0.0</td></tr>
    	<tr><td>21  </td><td> 1.0</td></tr>
    	<tr><td>22  </td><td> 1.5</td></tr>
    	<tr><td>23  </td><td> 4.0</td></tr>
    	<tr><td>24  </td><td> 8.5</td></tr>
    	<tr><td>25  </td><td>13.5</td></tr>
    	<tr><td>26  </td><td>18.0</td></tr>
    	<tr><td>27  </td><td>33.0</td></tr>
    	<tr><td>28  </td><td>67.0</td></tr>
    	<tr><td>29  </td><td>85.5</td></tr>
    	<tr><td>⋮</td><td>⋮</td></tr>
    	<tr><td> 71</td><td>0  </td></tr>
    	<tr><td> 72</td><td>0  </td></tr>
    	<tr><td> 73</td><td>0  </td></tr>
    	<tr><td> 74</td><td>0  </td></tr>
    	<tr><td> 75</td><td>0  </td></tr>
    	<tr><td> 76</td><td>0  </td></tr>
    	<tr><td> 77</td><td>0  </td></tr>
    	<tr><td> 78</td><td>0  </td></tr>
    	<tr><td> 79</td><td>0  </td></tr>
    	<tr><td> 80</td><td>0  </td></tr>
    	<tr><td> 81</td><td>0  </td></tr>
    	<tr><td> 82</td><td>0  </td></tr>
    	<tr><td> 83</td><td>0  </td></tr>
    	<tr><td> 84</td><td>0  </td></tr>
    	<tr><td> 85</td><td>0  </td></tr>
    	<tr><td> 86</td><td>0  </td></tr>
    	<tr><td> 87</td><td>0  </td></tr>
    	<tr><td> 88</td><td>0  </td></tr>
    	<tr><td> 89</td><td>0  </td></tr>
    	<tr><td> 90</td><td>0  </td></tr>
    	<tr><td> 91</td><td>0  </td></tr>
    	<tr><td> 92</td><td>0  </td></tr>
    	<tr><td> 93</td><td>0  </td></tr>
    	<tr><td> 94</td><td>0  </td></tr>
    	<tr><td> 95</td><td>0  </td></tr>
    	<tr><td> 96</td><td>0  </td></tr>
    	<tr><td> 97</td><td>0  </td></tr>
    	<tr><td> 98</td><td>0  </td></tr>
    	<tr><td> 99</td><td>0  </td></tr>
    	<tr><td>100</td><td>0  </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$per_base_n_content</dt>
    		<dd><table>
    <thead><tr><th scope=col>Base</th><th scope=col>N-Count</th></tr></thead>
    <tbody>
    	<tr><td>1    </td><td>0    </td></tr>
    	<tr><td>2    </td><td>0    </td></tr>
    	<tr><td>3    </td><td>0    </td></tr>
    	<tr><td>4    </td><td>0    </td></tr>
    	<tr><td>5    </td><td>0    </td></tr>
    	<tr><td>6    </td><td>0    </td></tr>
    	<tr><td>7    </td><td>0    </td></tr>
    	<tr><td>8    </td><td>0    </td></tr>
    	<tr><td>9    </td><td>0    </td></tr>
    	<tr><td>10-11</td><td>0    </td></tr>
    	<tr><td>12-13</td><td>0    </td></tr>
    	<tr><td>14-15</td><td>0    </td></tr>
    	<tr><td>16-17</td><td>0    </td></tr>
    	<tr><td>18-19</td><td>0    </td></tr>
    	<tr><td>20-21</td><td>0    </td></tr>
    	<tr><td>22-23</td><td>0    </td></tr>
    	<tr><td>24-25</td><td>0    </td></tr>
    	<tr><td>26-27</td><td>0    </td></tr>
    	<tr><td>28-29</td><td>0    </td></tr>
    	<tr><td>30-31</td><td>0    </td></tr>
    	<tr><td>32-33</td><td>0    </td></tr>
    	<tr><td>34-35</td><td>0    </td></tr>
    	<tr><td>36-37</td><td>0    </td></tr>
    	<tr><td>38-39</td><td>0    </td></tr>
    	<tr><td>40-41</td><td>0    </td></tr>
    	<tr><td>42-43</td><td>0    </td></tr>
    	<tr><td>44-45</td><td>0    </td></tr>
    	<tr><td>46-47</td><td>0    </td></tr>
    	<tr><td>48-49</td><td>0    </td></tr>
    	<tr><td>50-51</td><td>0    </td></tr>
    	<tr><td>52-53</td><td>0    </td></tr>
    	<tr><td>54-55</td><td>0    </td></tr>
    	<tr><td>56-57</td><td>0    </td></tr>
    	<tr><td>58-59</td><td>0    </td></tr>
    	<tr><td>60-61</td><td>0    </td></tr>
    	<tr><td>62-63</td><td>0    </td></tr>
    	<tr><td>64-65</td><td>0    </td></tr>
    	<tr><td>66-67</td><td>0    </td></tr>
    	<tr><td>68-69</td><td>0    </td></tr>
    	<tr><td>70-71</td><td>0    </td></tr>
    	<tr><td>72-73</td><td>0    </td></tr>
    	<tr><td>74-75</td><td>0    </td></tr>
    	<tr><td>76-77</td><td>0    </td></tr>
    	<tr><td>78-79</td><td>0    </td></tr>
    	<tr><td>80-81</td><td>0    </td></tr>
    	<tr><td>82-83</td><td>0    </td></tr>
    	<tr><td>84-85</td><td>0    </td></tr>
    	<tr><td>86-87</td><td>0    </td></tr>
    	<tr><td>88-89</td><td>0    </td></tr>
    	<tr><td>90-91</td><td>0    </td></tr>
    	<tr><td>92-93</td><td>0    </td></tr>
    	<tr><td>94-95</td><td>0    </td></tr>
    	<tr><td>96-97</td><td>0    </td></tr>
    	<tr><td>98-99</td><td>0    </td></tr>
    	<tr><td>100  </td><td>0    </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$sequence_length_distribution</dt>
    		<dd><table>
    <thead><tr><th scope=col>Length</th><th scope=col>Count</th></tr></thead>
    <tbody>
    	<tr><td>48-49  </td><td>   1   </td></tr>
    	<tr><td>50-51  </td><td>   0   </td></tr>
    	<tr><td>52-53  </td><td>   1   </td></tr>
    	<tr><td>54-55  </td><td>   0   </td></tr>
    	<tr><td>56-57  </td><td>   0   </td></tr>
    	<tr><td>58-59  </td><td>   1   </td></tr>
    	<tr><td>60-61  </td><td>   3   </td></tr>
    	<tr><td>62-63  </td><td>   0   </td></tr>
    	<tr><td>64-65  </td><td>   0   </td></tr>
    	<tr><td>66-67  </td><td>   4   </td></tr>
    	<tr><td>68-69  </td><td>   2   </td></tr>
    	<tr><td>70-71  </td><td>   1   </td></tr>
    	<tr><td>72-73  </td><td>   1   </td></tr>
    	<tr><td>74-75  </td><td>   4   </td></tr>
    	<tr><td>76-77  </td><td>   0   </td></tr>
    	<tr><td>78-79  </td><td>   3   </td></tr>
    	<tr><td>80-81  </td><td>   5   </td></tr>
    	<tr><td>82-83  </td><td>   4   </td></tr>
    	<tr><td>84-85  </td><td>   7   </td></tr>
    	<tr><td>86-87  </td><td>   1   </td></tr>
    	<tr><td>88-89  </td><td>   9   </td></tr>
    	<tr><td>90-91  </td><td>  16   </td></tr>
    	<tr><td>92-93  </td><td>  16   </td></tr>
    	<tr><td>94-95  </td><td>   0   </td></tr>
    	<tr><td>96-97  </td><td>   0   </td></tr>
    	<tr><td>98-99  </td><td>   0   </td></tr>
    	<tr><td>100-101</td><td>9051   </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$sequence_duplication_levels</dt>
    		<dd><table>
    <thead><tr><th scope=col>Duplication Level</th><th scope=col>Percentage of deduplicated</th><th scope=col>Percentage of total</th></tr></thead>
    <tbody>
    	<tr><td>1          </td><td>61.91268191</td><td>32.617744  </td></tr>
    	<tr><td>2          </td><td>20.72765073</td><td>21.840088  </td></tr>
    	<tr><td>3          </td><td> 9.20997921</td><td>14.556407  </td></tr>
    	<tr><td>4          </td><td> 3.24324324</td><td> 6.834611  </td></tr>
    	<tr><td>5          </td><td> 1.89189189</td><td> 4.983571  </td></tr>
    	<tr><td>6          </td><td> 0.95634096</td><td> 3.023001  </td></tr>
    	<tr><td>7          </td><td> 0.29106029</td><td> 1.073384  </td></tr>
    	<tr><td>8          </td><td> 0.37422037</td><td> 1.577218  </td></tr>
    	<tr><td>9          </td><td> 0.33264033</td><td> 1.577218  </td></tr>
    	<tr><td><span style=white-space:pre-wrap>&gt;10  </span></td><td> 0.99792100                                     </td><td> 8.105148                                       </td></tr>
    	<tr><td><span style=white-space:pre-wrap>&gt;50  </span></td><td> 0.04158004                                     </td><td> 1.161008                                       </td></tr>
    	<tr><td>&gt;100    </td><td> 0.02079002</td><td> 2.650602  </td></tr>
    	<tr><td>&gt;500    </td><td> 0.00000000</td><td> 0.000000  </td></tr>
    	<tr><td><span style=white-space:pre-wrap>&gt;1k  </span></td><td> 0.00000000                                     </td><td> 0.000000                                       </td></tr>
    	<tr><td><span style=white-space:pre-wrap>&gt;5k  </span></td><td> 0.00000000                                     </td><td> 0.000000                                       </td></tr>
    	<tr><td>&gt;10k+   </td><td> 0.00000000</td><td> 0.000000  </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$overrepresented_sequences</dt>
    		<dd><table>
    <thead><tr><th scope=col>Sequence</th><th scope=col>Count</th><th scope=col>Percentage</th><th scope=col>Possible Source</th></tr></thead>
    <tbody>
    	<tr><td>GCACCTAAAGACATATGTCCGACCTTGAAAAAAGGATTTTTGTGTGACAG</td><td>242                                               </td><td>2.6506024                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGCGAATTCACTAGTGATTTAATACGACTCACTATAGGGGAGGAAGATTA</td><td> 55                                               </td><td>0.6024096                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTGATTTAATACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTT</td><td> 51                                               </td><td>0.5585980                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CTGGGTATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAATTT</td><td> 29                                               </td><td>0.3176342                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCT</td><td> 26                                               </td><td>0.2847755                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>ATTTAATACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTATT</td><td> 24                                               </td><td>0.2628697                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GCGAATTCACTAGTGATTTAATACGACTCACTATAGGGGAGGAAGATTAA</td><td> 24                                               </td><td>0.2628697                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GAAAAACTGCTTATTGGGTCTTTCCGTGTTTAAGATGGAGCAGTTGAAAT</td><td> 23                                               </td><td>0.2519168                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGGGTATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAATTTC</td><td> 22                                               </td><td>0.2409639                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGCTTATTGGGTCTTTCCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCT</td><td> 22                                               </td><td>0.2409639                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTATTGGGTCTTT</td><td> 22                                               </td><td>0.2409639                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTATTGGGTCTTTCCG</td><td> 21                                               </td><td>0.2300110                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACA</td><td> 21                                               </td><td>0.2300110                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TCACTAGTGATTTAATACGACTCACTATAGGGGAGGAAGATTAAGAAAAA</td><td> 21                                               </td><td>0.2300110                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TAGTGATTTAATACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGC</td><td> 21                                               </td><td>0.2300110                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>ATTTAATATCAAGAGGAAGAATTTCAACTGCTCCATCTTAAACACGGAAA</td><td> 20                                               </td><td>0.2190581                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GCGGCCGCGAATTCACTAGTGATTTAATACGACTCACTATAGGGGAGGAA</td><td> 20                                               </td><td>0.2190581                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GGAAGATTAAGAAAAACTGCTTATTGGGTCTTTCCGTGTTTTAGATGAAG</td><td> 20                                               </td><td>0.2190581                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCTCCATCTTAA</td><td> 19                                               </td><td>0.2081051                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CAGCTCCTCGCCCTTGCTCACGCGTCTGGGTATTGGGTATGTTGTGTAGC</td><td> 17                                               </td><td>0.1861993                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTCTGGGTATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAAT</td><td> 16                                               </td><td>0.1752464                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GGAAGATTAAGAAAAACTGCTTATTGGGTCTTTCCGTGTTTAAGATGGAG</td><td> 15                                               </td><td>0.1642935                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTTTTAGATGAAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACA</td><td> 15                                               </td><td>0.1642935                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>AGAAAAACTGCTTATTGGGTCTTTCCGTGTTTTAGATGAAGCAGTTGAAA</td><td> 14                                               </td><td>0.1533406                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTTGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCTCCATCTT</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CTCCTCGCCCTTGCTCACGCGTCTGGGTATTGGGTATGTTGTGTAGCCAT</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GATTTAATACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTAT</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCTTCATCTAAAACAC</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTTGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCTTCATCTA</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GAAAAACTGCTTATTGGGTCTTTCCGTGTTTTAGATGAAGCAGTTGAAAT</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>ATTTAATATCAAGAGGAAGAATTTCAACTGCTTCATCTAAAACACGGAAA</td><td> 13                                               </td><td>0.1423877                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGGCCGCGAATTCACTAGTGATTTAATACGACTCACTATAGGGGAGGAAG</td><td> 12                                               </td><td>0.1314348                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGGGTCTTTCCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATA</td><td> 12                                               </td><td>0.1314348                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGAATTCACTAGTGATTTAATACGACTCACTATAGGGGAGGAAGATTAAG</td><td> 12                                               </td><td>0.1314348                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGTCTGGGTATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAA</td><td> 12                                               </td><td>0.1314348                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGC</td><td> 12                                               </td><td>0.1314348                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTATTGGGTCT</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTGCTTCATCTAAA</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTTCAAGAAATACTTCAAATGACCCCTTCACATTACTCAGGAAATATTGT</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GGATTCTCGTCCTCAGAAAATCTGGATGGCGCCGAGTCTCACTGAATCTG</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>AACTGCTCCATCTTAAACACGGAAAGACCCAATAAGCAGTTTTTCTTAAT</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>AGAAAAACTGCTTATTGGGTCTTTCCGTGTTTAAGATGGAGCAGTTGAAA</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CCTCGCCCTTGCTCACGCGTCTGGGTATTGGGTATGTTGTGTAGCCATTT</td><td> 11                                               </td><td>0.1204819                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TGGCGATACATCTCCTCAAAAGAGTGTTCTGACTGGGTGTTGTCACTGTC</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>GTCTTTCCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATATTA</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>ATACGACTCACTATAGGGGAGGAAGATTAAGAAAAACTGCTTATTGGGTC</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CCTGGTTCCTGGCCTCTTGAGTGTGGTCCTGATCTTGTTGCTCGTCTTGC</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>AAGAATTTCAACTGCTCCATCTTAAACACGGAAAGACCCAATAAGCAGTT</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>ATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAGAATTTCAACTG</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>CGCGTCTGGGTATTGGGTATGTTGTGTAGCCATTTAATATCAAGAGGAAG</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    	<tr><td>TTCAACTGCTCCATCTTAAACACGGAAAGACCCAATAAGCAGTTTTTCTT</td><td> 10                                               </td><td>0.1095290                                         </td><td>No Hit                                            </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$adapter_content</dt>
    		<dd><table>
    <thead><tr><th scope=col>Position</th><th scope=col>Illumina Universal Adapter</th><th scope=col>Illumina Small RNA 3' Adapter</th><th scope=col>Illumina Small RNA 5' Adapter</th><th scope=col>Nextera Transposase Sequence</th><th scope=col>SOLID Small RNA Adapter</th></tr></thead>
    <tbody>
    	<tr><td>1    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>2    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>3    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>4    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>5    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>6    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>7    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>8    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>9    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>10-11</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>12-13</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>14-15</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>16-17</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>18-19</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>20-21</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>22-23</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>24-25</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>26-27</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>28-29</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>30-31</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>32-33</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>34-35</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>36-37</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>38-39</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>40-41</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>42-43</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>44-45</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>46-47</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>48-49</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>50-51</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>52-53</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>54-55</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>56-57</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>58-59</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>60-61</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>62-63</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>64-65</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>66-67</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>68-69</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>70-71</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>72-73</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>74-75</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>76-77</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>78-79</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>80-81</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>82-83</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>84-85</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>86-87</td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    	<tr><td>88   </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td><td>0    </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$kmer_content</dt>
    		<dd><table>
    <thead><tr><th scope=col>Sequence</th><th scope=col>Count</th><th scope=col>PValue</th><th scope=col>Obs/Exp Max</th><th scope=col>Max Obs/Exp Position</th></tr></thead>
    <tbody>
    	<tr><td>AGGGAAT     </td><td>15          </td><td>0.0006416716</td><td>94.95557    </td><td>94          </td></tr>
    	<tr><td>CCTAAAG     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>4           </td></tr>
    	<tr><td>CACCTAA     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>2           </td></tr>
    	<tr><td>AAGACAT     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>8           </td></tr>
    	<tr><td>CTAAAGA     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>5           </td></tr>
    	<tr><td>GCACCTA     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>1           </td></tr>
    	<tr><td>ACCTAAA     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>3           </td></tr>
    	<tr><td>AGACATA     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>9           </td></tr>
    	<tr><td>TAAAGAC     </td><td>15          </td><td>0.0006707299</td><td>93.91209    </td><td>6           </td></tr>
    	<tr><td>CAGCTCC     </td><td>20          </td><td>0.0021022684</td><td>70.43407    </td><td>1           </td></tr>
    	<tr><td>CAATTGA     </td><td>20          </td><td>0.0021022684</td><td>70.43407    </td><td>9           </td></tr>
    	<tr><td>TAACTGA     </td><td>25          </td><td>0.0049792626</td><td>56.65856    </td><td>86-87       </td></tr>
    	<tr><td>AAAGACA     </td><td>25          </td><td>0.0050899666</td><td>56.34725    </td><td>7           </td></tr>
    	<tr><td>TGAAAAA     </td><td>30          </td><td>0.0040738895</td><td>31.30403    </td><td>26-27       </td></tr>
    	<tr><td>TTGAAAA     </td><td>35          </td><td>0.0086529500</td><td>26.83203    </td><td>24-25       </td></tr>
    </tbody>
    </table>
    </dd>
    	<dt>$total_deduplicated_percentage</dt>
    		<dd>52.68</dd>
    </dl>



Gyenge minőségű nukleotidok, readek kiszűrése
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

trimming, filtering

.. code:: ipython3

    # R
    library(seqTools)
    
    fajlom = 'SRR1660259.fastq'
    
    trimFastq(
      infile        = fajlom, 
      outfile       = 'marad.fq.gz',
      discard       = 'kimarad.fq.gz',
      qualDiscard   = 10, # All reads which contain one or more phred scores < qualDiscard will be discarded.
      fixTrimLeft   = 0,  # Prefix of this size will be trimmed.
      fixTrimRight  = 0,  # Suffix of this size will be trimmed.
      qualTrimLeft  = 30, # Prefix where all phred scores are < qualTrimLeft will be trimmed.
      qualTrimRight = 30, # Suffix where all phred scores are < qualTrimRight will be trimmed.
      minSeqLen     = 50  # All reads where sequence length after (fixed and quality based) trimming is < minSeqLen will be discarded.
    )

.. code:: ipython3

    library(seqTools)
    
    ?trimFastq



.. parsed-literal::

    Loading required package: zlibbioc


Readek illesztése referencia-genomra
------------------------------------

.. code:: bash

    %%bash 
    
    cd gyak06
    
    export PATH=$PATH:/home/bioinfo/edirect
    
    efetch -db=nuccore -format=fasta -id=AF086833 > ebola1976.fa
    
    ls -lh


.. parsed-literal::

    total 4.0M
    -rw-rw-r-- 1 sn sn  19K Mar 20 10:48 ebola1976.fa
    drwxrwxr-x 2 sn sn 4.0K Mar 20 09:35 FASTQC
    -rwxrwxr-x 1 sn sn  13K Mar 20 09:39 minta_jelentes.Rmd
    -rw-rw-r-- 1 sn sn 1.6M Mar 20 09:46 osszefoglalo.html
    -rw-rw-r-- 1 sn sn 2.4M Mar 20 09:31 SRR1660259.fastq


.. code:: bash

    %%bash 
    
    cd gyak06
    
    head ebola1976.fa


.. parsed-literal::

    >AF086833.2 Ebola virus - Mayinga, Zaire, 1976, complete genome
    CGGACACACAAAAAGAAAGAAGAATTTTTAGGATCTTTTGTGTGCGAATAACTATGAGGAAGATTAATAA
    TTTTCCTCTCATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCACACCTGGTTTGTTT
    CAGAGCCACATCACAAAGATAGAGAACAACCTAGGTCTCCGAAGGGAGCAAGGGCATCAGTGTGCTCAGT
    TGAAAATCCCTTGTCAACACCTAGGTCTTATCACATCACAAGTTCCACCTCAGACTCTGCAGGGTGATCC
    AACAACCTTAATAGAAACATTATTGTTAAAGGACAGCATTAGTTCACAGTCAAACAAGCAAGATTGAGAA
    TTAACCTTGGTTTTGAACTTGAACACTTAGGGGATTGAAGATTCAACAACCCTAAAGCTTGGGGTAAAAC
    ATTGGAAATAGTTAAAAGACAAATTGCTCGGAATCACAAAATTCCGAGTATGGATTCTCGTCCTCAGAAA
    ATCTGGATGGCGCCGAGTCTCACTGAATCTGACATGGATTACCACAAGATCTTGACAGCAGGTCTGTCCG
    TTCAACAGGGGATTGTTCGGCAAAGAGTCATCCCAGTGTATCAAGTAAACAATCTTGAAGAAATTTGCCA


.. code:: bash

    %%bash 
    
    # http://bio-bwa.sourceforge.net/
    
    cd gyak06
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa index -p Ebola ebola1976.fa
    
    ls -lh


.. parsed-literal::

    total 4.0M
    -rw-rw-r-- 1 sn sn  19K Mar 20 10:48 ebola1976.fa
    -rw-rw-r-- 1 sn sn   10 Mar 20 10:53 Ebola.amb
    -rw-rw-r-- 1 sn sn   86 Mar 20 10:53 Ebola.ann
    -rw-rw-r-- 1 sn sn  19K Mar 20 10:53 Ebola.bwt
    -rw-rw-r-- 1 sn sn 4.7K Mar 20 10:53 Ebola.pac
    -rw-rw-r-- 1 sn sn 9.4K Mar 20 10:53 Ebola.sa
    drwxrwxr-x 2 sn sn 4.0K Mar 20 09:35 FASTQC
    -rwxrwxr-x 1 sn sn  13K Mar 20 09:39 minta_jelentes.Rmd
    -rw-rw-r-- 1 sn sn 1.6M Mar 20 09:46 osszefoglalo.html
    -rw-rw-r-- 1 sn sn 2.4M Mar 20 09:31 SRR1660259.fastq


.. parsed-literal::

    [bwa_index] Pack FASTA... 0.00 sec
    [bwa_index] Construct BWT for the packed sequence...
    [bwa_index] 0.01 seconds elapse.
    [bwa_index] Update BWT... 0.00 sec
    [bwa_index] Pack forward-only FASTA... 0.00 sec
    [bwa_index] Construct SA from BWT and Occ... 0.00 sec
    [main] Version: 0.7.17-r1188
    [main] CMD: bwa index -p Ebola ebola1976.fa
    [main] Real time: 0.183 sec; CPU: 0.012 sec


.. code:: bash

    %%bash 
    
    cd gyak06
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa mem Ebola SRR1660259.fastq > illesztes01.sam
    
    ls -lh


.. parsed-literal::

    total 6.5M
    -rw-rw-r-- 1 sn sn  19K Mar 20 10:48 ebola1976.fa
    -rw-rw-r-- 1 sn sn   10 Mar 20 10:53 Ebola.amb
    -rw-rw-r-- 1 sn sn   86 Mar 20 10:53 Ebola.ann
    -rw-rw-r-- 1 sn sn  19K Mar 20 10:53 Ebola.bwt
    -rw-rw-r-- 1 sn sn 4.7K Mar 20 10:53 Ebola.pac
    -rw-rw-r-- 1 sn sn 9.4K Mar 20 10:53 Ebola.sa
    drwxrwxr-x 2 sn sn 4.0K Mar 20 09:35 FASTQC
    -rw-rw-r-- 1 sn sn 2.5M Mar 20 10:57 illesztes01.sam
    -rwxrwxr-x 1 sn sn  13K Mar 20 09:39 minta_jelentes.Rmd
    -rw-rw-r-- 1 sn sn 1.6M Mar 20 09:46 osszefoglalo.html
    -rw-rw-r-- 1 sn sn 2.4M Mar 20 09:31 SRR1660259.fastq


.. parsed-literal::

    [M::bwa_idx_load_from_disk] read 0 ALT contigs
    [M::process] read 9130 sequences (911661 bp)...
    [M::mem_process_seqs] Processed 9130 reads in 0.219 CPU sec, 0.218 real sec
    [main] Version: 0.7.17-r1188
    [main] CMD: bwa mem Ebola SRR1660259.fastq
    [main] Real time: 0.313 sec; CPU: 0.236 sec


SAM-fájl
~~~~~~~~

Az illesztés eredményét tartalmazó tabulátorral osztott fájl (CSV), az
egyes részek nem azonos oszlopszámúak
(https://samtools.github.io/hts-specs/SAMv1.pdf).

.. code:: bash

    %%bash 
    
    cd gyak06
    
    head illesztes01.sam


.. parsed-literal::

    @SQ	SN:AF086833.2	LN:18959
    @PG	ID:bwa	PN:bwa	VN:0.7.17-r1188	CL:bwa mem Ebola SRR1660259.fastq
    SRR1660259.1	16	AF086833.2	7706	60	100M	*	0	0	GGGTTGAGACAGCTGGCCAACGAGACGACTCAAGCTCTTCAACTGTTCCTGAGAGCCACAACTGAGCTACGCACCTTTTCAATCCTCAACCGTAAGGCAA	>@CCCA>@5C>??3CA?9==CAAACA@@FEDCEHEHHGEHGEGFEGBIIHEFEIGHF?BFDGHHCIGGHBGCHIGIIIGGHFC+GBAFFHDHFFFFF@@@	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.2	0	AF086833.2	814	60	100M	*	0	0	GCGCCTTGAGGAATTGCTGCCAGCAGTATCTAGTGGAAAAAACATTAAGAGAACACTTGCTGCCATGCCGGAAGAGGAGACAACTGAAGCTAATGCCGGT	CCCFFFFFHHHHHJJJJJJJJJJJJJHIJJJJJHHIJJJJIJJJJJIJJJJGIIJJJJJJJJJJJJHHHHFDDDDDDDDDDDDDDDDDDDDDDDDDDDDB	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.3	0	AF086833.2	11530	60	83M17S	*	0	0	GTCTTTCCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACACAACATACCCAATACCCAGACGCGTGAGCAAGGGCGAGGA	@CCDF?EFHHHDHJJJJEIJJIGGIJHIIJIIJJJGGHJJFGIIDGIJIIIHIIGEGIGGHJIJIIGHHHJGIG:CAEEDFF@DDDDDDDDDDDDDDD@9	NM:i:2	MD:Z:14T5A62	AS:i:73	XS:i:0
    SRR1660259.4	16	AF086833.2	1065	60	100M	*	0	0	TGATTTTCCGTTTGATGCGAACAAATTTTCTGATCAAATTTCTCCTAATACACCAAGGGATGCACATGGTTGCCGGGCATGATGCCAACGATGCTGTGAT	DDDDDDDDDDDCDDDDDDDDCEEEEEFFFFFHHHHHHHGIHJJJIIJIHCGCJJIIJJJJJJIGJJJJIJJJJJIJJJJJJJJJJJJHHHHHFFFFFCCC	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.5	0	AF086833.2	11510	60	100M	*	0	0	TAAGAAAAACTGCTTATTGGGTCTTTCCGTGTTTTAGATGAAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACACAACATACCCAATACCCAGA	CCCFFFFFHHHHHJJJJJJJJFHJJJJJJGHHJJJJJIJJIJJJJJIIJJJJJIJIIJJJJJJJJJJJJJIJJIJIHHGHHFFFFEEEEEDDDDDDDDDD	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.6	0	AF086833.2	2660	60	100M	*	0	0	TTCATGGCAATCCTGCAACATCATCAGTGAATGAGCATGGAACAATGGGATGATTCAACCGACAAATAGCTAACATTAAGTAGTCAAGGAACGAAAACAG	CCCFFFFFHHHHHJJJJJJJJJJJJJJHIJJJJJJJJJJJIJJJJJJJJJJJIJJJJJJJJJJJHHHHHFFFFFFEEEEEDEEFEEDDDDDDDDDDDDDD	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.7	16	AF086833.2	7235	60	100M	*	0	0	GAGGCAACTCAAGTTGAACAACATCCCCGCAGAACAGACAACGACAGCACAGCCTCCGACACTCCCTCTGCCACGACCGCAGCCGGACCCCCAAAAGCAG	ACDDDDCC@CCDDEDCDCCCC<<<5&B@ACCAC@:C>?<BDDC@CCA>8?DDDDDDDEA>5<<FHC;HGJIIIIGFJJJJJJJJJJJHHHFHFFFFFCCC	NM:i:1	MD:Z:25A74	AS:i:95	XS:i:0
    SRR1660259.8	16	AF086833.2	3969	60	100M	*	0	0	ACAAAAAGAGTTCCAATCTTCCAAGATGCTGCTCCACCTGTCATCCACATCCGCTCTCGAGGTGACATTCCCCGAGCTTGCCAGAAAAGCTTGCGTCCAG	<BDDDDDDDDDDDDDDBCDDDDDDCDDBDBDDBDCCDDDDDDDDDDCDDDFHHJJJIJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJHHHHHFFFFFCCC	NM:i:0	MD:Z:100	AS:i:100	XS:i:0


Fejléc:

::

    @SQ SN:AF086833.2   LN:18959
    @PG ID:bwa  PN:bwa  VN:0.7.17-r1188 CL:bwa mem ../genomes/Ebola SRR1660257.fastq

Illesztési eredmény 1. sora:

::

    SRR1660257.1    16  AF086833.2  8058    60  100M    *   0   0TTGTCTTTTAGTTTTTCTTCAGATTGCTTCATGGAAAAGCTCAGCCTCAAATCAATGAAACCAGGATTTAATTATATGGATTACTTGAATCTAAGATTAC   DDDDCEECFFFFHHHGHHJJJJJJIJJJJJJJJJJJJJJJIJJJJJIJJJJJJJJJJJIGJJJJJJJJJJJJJJJJJJJJJIHJIJJHHHHHFFFFFCBC    NM:i:0  MD:Z:100    AS:i:100XS:i:0

Kötelező elemei:

::

    QNAME = SRR1660257.1 # Query template NAME
    FLAG  = 16           # bitwise FLAG; 16=SEQ being reverse complemented
    RNAME = AF086833.2   # Reference sequence NAME
    POS   = 8058         # 1-based leftmost mapping POSition
    MAPQ  = 60           # MAPping Quality (https://genome.sph.umich.edu/wiki/Mapping_Quality_Scores)
    CIGAR = 100M         # CIGAR string
    RNEXT = *            # Ref. name of the mate/next read
    PNEXT = 0            # Position of the mate/next read
    TLEN  = 0            # observed Template LENgth
    SEQ   = TTGTCTTTTAGTTTTTCTTCAGATTGCTTCATGGAAAAGCTCAGCCTCAAATCAATGAAACCAGGATTTAATTATATGGATTACTTGAATCTAAGATTAC
    QUAL  = DDDDCEECFFFFHHHGHHJJJJJJIJJJJJJJJJJJJJJJIJJJJJIJJJJJJJJJJJIGJJJJJJJJJJJJJJJJJJJJJIHJIJJHHHHHFFFFFCBC

Opcionális elemei (http://samtools.github.io/hts-specs/SAMtags.pdf):

::

    NM:i:0              # Edit distance to the reference, including ambiguous bases but excluding clipping
    MD:Z:100            # String for mismatching positions, the field ought to match the CIGAR string.
    AS:i:100            # Alignment score generated by aligner
    XS:i:0              # Reserved for end users

A MAPQ azt fejezi ki PHRED-pontszámmal, hogy a read milyen
valószínűséggel lett hibásan illesztve.

CIGAR string értelmezéséhez:

::

    M   alignment match (can be a sequence match or mismatch)
    I   insertion to the reference
    D   deletion from the reference
    N   skipped region from the reference
    S   soft clipping (clipped sequences present in SEQ)
    H   hard clipping (clipped sequences NOT present in SEQ)
    P   padding (silent deletion from padded reference)
    =   sequence match
    X   sequence mismatch

BAM-fájl
~~~~~~~~

Bináris, tömörített SAM-fájl. A kisebb mérete mellett nagy előnye, hogy
lekérdezhető, kisebb részletek kiolvashatók belőle anélkül, hogy a
teljes állományt be kellene tölteni a memóriába.

.. code:: bash

    %%bash 
    
    cd gyak06
    
    samtools view -Sb illesztes01.sam > illesztes01.bam

.. code:: bash

    %%bash
    
    cd gyak06
    
    samtools sort illesztes01.bam > illesztes01_sorted.bam
    samtools index illesztes01_sorted.bam

.. code:: ipython3

    # TERMINÁLBAN !!!
    
    cd gyak06
    
    samtools tview illesztes01_sorted.bam
    
    samtools tview illesztes01_sorted.bam ebola1976.fa


-  ?: súgó ablakot nyit
-  .: forward illesztésű, a referenciával megegyező nukleotid
-  ,: reverse illesztésű, a referenciával megegyező nukleotid
-  ACGT: forward illesztésű, referenciától eltérő nukleotid
-  acgt: reverse illesztésű, referenciától eltérő nukleotid
-  \*: törölt bázis

.. code:: bash

    %%bash
    
    cd gyak06
    
    samtools view -Sb -F 4 illesztes01.sam > illesztes01_van.bam 
    
    samtools view -Sb -f 4 illesztes01.sam > illesztes01_nincs.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak06
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa mem Ebola SRR1660259.fastq | samtools view -Sb -F 4 > illesztes02_van.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak06
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa mem Ebola SRR1660259.fastq | samtools view -Sb -F 4 | samtools sort > illesztes03_van_sorted.bam
    
    samtools index illesztes03_van_sorted.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak06
    
    # samtools mpileup illesztes03_van_sorted.bam > illesztes03_van_sorted.pileup
    
    samtools mpileup -f ebola1976.fa illesztes03_van_sorted.bam > illesztes03_van_sorted.pileup
    
    tail illesztes03_van_sorted.pileup

Oszlopok jelentése: 1. a referencia szekvencia azonosítója 2. a
szekvencia nukleotidjának pozíciója (1-ről induló sorszám) 3. a
referencia nukleotid az adott pozíción 4. az adott pozíciót lefedő
readek száma 5. az illesztett readek nukleotidja az adott pozícióban 6.
az illesztett readek adott pozícióbeli nukleotidjának minősége

További információ: https://en.wikipedia.org/wiki/Pileup\_format

.. code:: ipython3

    # R
    library(Rsamtools)
    
    bam.fajlom = 'illesztes03_van_sorted.bam'
    
    countBam(bam.fajlom)

.. code:: ipython3

    # R
    idxstatsBam(bam.fajlom)

.. code:: ipython3

    # R
    # átlagos lefedettség
    
    ref.szek.hossza = idxstatsBam(bam.fajlom)$seqlength
    illesztett.nukleotidok.szama = countBam(bam.fajlom)$nucleotides
    
    illesztett.nukleotidok.szama / ref.szek.hossza

.. code:: ipython3

    # R
    # pileup
    
    olvasasi.parameterek = PileupParam(max_depth=50)
    
    my.pileup = pileup(bam.fajlom, pileupParam=olvasasi.parameterek)
    
    head(my.pileup)


.. code:: ipython3

    # R
    # http://ggplot2.org/
    
    library(ggplot2)
    
    ggplot(data=my.pileup, aes(x=pos, y=count)) + geom_point()


.. code:: ipython3

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count)) + geom_point() + theme_bw()

.. code:: ipython3

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count, color=strand)) + geom_point() + theme_bw()

.. code:: ipython3

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count, color=strand)) + geom_point() + theme_bw() + geom_smooth()

.. code:: ipython3

    # R
    library(Gviz)
    
    # https://www.bioconductor.org/packages/release/bioc/html/Gviz.html
    # https://www.bioconductor.org/packages/release/bioc/vignettes/Gviz/inst/doc/Gviz.pdf
    
    options(ucscChromosomeNames=FALSE)
    # https://www.ncbi.nlm.nih.gov/nuccore/KM034562.1
    # http://hgdownload.cse.ucsc.edu/downloads.html#ebola_virus
    
    illesztes.track = AlignmentsTrack(bam.fajlom, start=2000, end=3000)
    
    plotTracks(
      list(illesztes.track), 
      type=c('coverage', 'pileup'), 
      chromosome='AF086833.2', 
      from=2000, 
      to=3000
    )

.. code:: ipython3

    # R
    tengely.track = GenomeAxisTrack()
    
    plotTracks(
      list(tengely.track, illesztes.track), 
      type=c('coverage', 'pileup'), 
      chromosome='AF086833.2', 
      from=2000, 
      to=3000
    )

.. code:: ipython3

    # R
    library(rentrez)
    library(seqinr)
    
    ebola = entrez_fetch(db='nuccore', id='AF086833.2', rettype='fasta')
    referencia = read.fasta(textConnection(ebola), as.string=TRUE, seqonly=TRUE)
    referencia.szekvencia = referencia[[1]]
    referencia.szekvencia = DNAStringSet(referencia.szekvencia)
    names(referencia.szekvencia) = 'AF086833.2'
    
    szekvencia.track = SequenceTrack(referencia.szekvencia)
    
    plotTracks(
      trackList = list(tengely.track, illesztes.track, szekvencia.track), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2000, 
      to = 3000
    )

.. code:: ipython3

    # R
    kiemeles = HighlightTrack(
      trackList = list(illesztes.track, szekvencia.track),
      chromosome = 'AF086833.2',
      start = 2218,
      end = 2222
    )
    
    plotTracks(
      list(tengely.track, kiemeles), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2200, 
      to = 2240
    )

.. code:: ipython3

    # R
    displayPars(kiemeles)$fill = 'blue'
    displayPars(kiemeles)$col = 'transparent'
    displayPars(kiemeles)$alpha = 0.3
                
    plotTracks(
      list(tengely.track, kiemeles), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2200, 
      to = 2240
    )
