
NGS III.
========

.. code:: bash

    %%bash
    
    export PATH=$PATH:/usr/local/ncbi/sra-tools/bin
    export PATH=$PATH:/home/bioinfo/edirect
    export PATH=$PATH:/home/bioinfo/bwa
    
    mkdir gyak07
    cd gyak07
    
    fastq-dump SRR1660259
    
    efetch -db=nuccore -format=fasta -id=AF086833 > ebola1976.fa
    
    bwa index -p Ebola ebola1976.fa
    bwa mem Ebola SRR1660259.fastq > illesztes01.sam

Sequence Alignment Map (SAM) fájl
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Az illesztés eredményét tartalmazó tabulátorral osztott fájl (CSV), az
egyes részek nem azonos oszlopszámúak
(https://samtools.github.io/hts-specs/SAMv1.pdf).

.. code:: bash

    %%bash 
    
    cd gyak07
    
    head illesztes01.sam


.. parsed-literal::

    @SQ	SN:AF086833.2	LN:18959
    @PG	ID:bwa	PN:bwa	VN:0.7.17-r1188	CL:bwa mem Ebola SRR1660259.fastq
    SRR1660259.1	16	AF086833.2	7706	60	100M	*	0	0	GGGTTGAGACAGCTGGCCAACGAGACGACTCAAGCTCTTCAACTGTTCCTGAGAGCCACAACTGAGCTACGCACCTTTTCAATCCTCAACCGTAAGGCAA	>@CCCA>@5C>??3CA?9==CAAACA@@FEDCEHEHHGEHGEGFEGBIIHEFEIGHF?BFDGHHCIGGHBGCHIGIIIGGHFC+GBAFFHDHFFFFF@@@	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.2	0	AF086833.2	814	60	100M	*	0	0	GCGCCTTGAGGAATTGCTGCCAGCAGTATCTAGTGGAAAAAACATTAAGAGAACACTTGCTGCCATGCCGGAAGAGGAGACAACTGAAGCTAATGCCGGT	CCCFFFFFHHHHHJJJJJJJJJJJJJHIJJJJJHHIJJJJIJJJJJIJJJJGIIJJJJJJJJJJJJHHHHFDDDDDDDDDDDDDDDDDDDDDDDDDDDDB	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.3	0	AF086833.2	11530	60	83M17S	*	0	0	GTCTTTCCGTGTTTAAGATGGAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACACAACATACCCAATACCCAGACGCGTGAGCAAGGGCGAGGA	@CCDF?EFHHHDHJJJJEIJJIGGIJHIIJIIJJJGGHJJFGIIDGIJIIIHIIGEGIGGHJIJIIGHHHJGIG:CAEEDFF@DDDDDDDDDDDDDDD@9	NM:i:2	MD:Z:14T5A62	AS:i:73	XS:i:0
    SRR1660259.4	16	AF086833.2	1065	60	100M	*	0	0	TGATTTTCCGTTTGATGCGAACAAATTTTCTGATCAAATTTCTCCTAATACACCAAGGGATGCACATGGTTGCCGGGCATGATGCCAACGATGCTGTGAT	DDDDDDDDDDDCDDDDDDDDCEEEEEFFFFFHHHHHHHGIHJJJIIJIHCGCJJIIJJJJJJIGJJJJIJJJJJIJJJJJJJJJJJJHHHHHFFFFFCCC	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.5	0	AF086833.2	11510	60	100M	*	0	0	TAAGAAAAACTGCTTATTGGGTCTTTCCGTGTTTTAGATGAAGCAGTTGAAATTCTTCCTCTTGATATTAAATGGCTACACAACATACCCAATACCCAGA	CCCFFFFFHHHHHJJJJJJJJFHJJJJJJGHHJJJJJIJJIJJJJJIIJJJJJIJIIJJJJJJJJJJJJJIJJIJIHHGHHFFFFEEEEEDDDDDDDDDD	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.6	0	AF086833.2	2660	60	100M	*	0	0	TTCATGGCAATCCTGCAACATCATCAGTGAATGAGCATGGAACAATGGGATGATTCAACCGACAAATAGCTAACATTAAGTAGTCAAGGAACGAAAACAG	CCCFFFFFHHHHHJJJJJJJJJJJJJJHIJJJJJJJJJJJIJJJJJJJJJJJIJJJJJJJJJJJHHHHHFFFFFFEEEEEDEEFEEDDDDDDDDDDDDDD	NM:i:0	MD:Z:100	AS:i:100	XS:i:0
    SRR1660259.7	16	AF086833.2	7235	60	100M	*	0	0	GAGGCAACTCAAGTTGAACAACATCCCCGCAGAACAGACAACGACAGCACAGCCTCCGACACTCCCTCTGCCACGACCGCAGCCGGACCCCCAAAAGCAG	ACDDDDCC@CCDDEDCDCCCC<<<5&B@ACCAC@:C>?<BDDC@CCA>8?DDDDDDDEA>5<<FHC;HGJIIIIGFJJJJJJJJJJJHHHFHFFFFFCCC	NM:i:1	MD:Z:25A74	AS:i:95	XS:i:0
    SRR1660259.8	16	AF086833.2	3969	60	100M	*	0	0	ACAAAAAGAGTTCCAATCTTCCAAGATGCTGCTCCACCTGTCATCCACATCCGCTCTCGAGGTGACATTCCCCGAGCTTGCCAGAAAAGCTTGCGTCCAG	<BDDDDDDDDDDDDDDBCDDDDDDCDDBDBDDBDCCDDDDDDDDDDCDDDFHHJJJIJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJHHHHHFFFFFCCC	NM:i:0	MD:Z:100	AS:i:100	XS:i:0


Fejléc:

::

    @SQ SN:AF086833.2   LN:18959
    @PG ID:bwa  PN:bwa  VN:0.7.17-r1188 CL:bwa mem ../genomes/Ebola SRR1660257.fastq

Illesztési eredmény 1. sora:

::

    SRR1660257.1    16  AF086833.2  8058    60  100M    *   0   0TTGTCTTTTAGTTTTTCTTCAGATTGCTTCATGGAAAAGCTCAGCCTCAAATCAATGAAACCAGGATTTAATTATATGGATTACTTGAATCTAAGATTAC   DDDDCEECFFFFHHHGHHJJJJJJIJJJJJJJJJJJJJJJIJJJJJIJJJJJJJJJJJIGJJJJJJJJJJJJJJJJJJJJJIHJIJJHHHHHFFFFFCBC    NM:i:0  MD:Z:100    AS:i:100XS:i:0

Kötelező elemei:

::

    QNAME = SRR1660257.1 # Query template NAME
    FLAG  = 16           # bitwise FLAG; 16=SEQ being reverse complemented
    RNAME = AF086833.2   # Reference sequence NAME
    POS   = 8058         # 1-based leftmost mapping POSition
    MAPQ  = 60           # MAPping Quality (https://genome.sph.umich.edu/wiki/Mapping_Quality_Scores)
    CIGAR = 100M         # CIGAR string
    RNEXT = *            # Ref. name of the mate/next read
    PNEXT = 0            # Position of the mate/next read
    TLEN  = 0            # observed Template LENgth
    SEQ   = TTGTCTTTTAGTTTTTCTTCAGATTGCTTCATGGAAAAGCTCAGCCTCAAATCAATGAAACCAGGATTTAATTATATGGATTACTTGAATCTAAGATTAC
    QUAL  = DDDDCEECFFFFHHHGHHJJJJJJIJJJJJJJJJJJJJJJIJJJJJIJJJJJJJJJJJIGJJJJJJJJJJJJJJJJJJJJJIHJIJJHHHHHFFFFFCBC

Opcionális elemei (http://samtools.github.io/hts-specs/SAMtags.pdf):

::

    NM:i:0              # Edit distance to the reference, including ambiguous bases but excluding clipping
    MD:Z:100            # String for mismatching positions, the field ought to match the CIGAR string.
    AS:i:100            # Alignment score generated by aligner
    XS:i:0              # Reserved for end users

A MAPQ azt fejezi ki PHRED-pontszámmal, hogy a read milyen
valószínűséggel lett hibásan illesztve.

CIGAR string értelmezéséhez:

::

    M   alignment match (can be a sequence match or mismatch)
    I   insertion to the reference
    D   deletion from the reference
    N   skipped region from the reference
    S   soft clipping (clipped sequences present in SEQ)
    H   hard clipping (clipped sequences NOT present in SEQ)
    P   padding (silent deletion from padded reference)
    =   sequence match
    X   sequence mismatch

Binary Alignment Map (BAM) fájl
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Bináris, tömörített SAM-fájl. A kisebb mérete mellett nagy előnye, hogy
lekérdezhető, kisebb részletek kiolvashatók belőle anélkül, hogy a
teljes állományt be kellene tölteni a memóriába.

.. code:: bash

    %%bash 
    
    cd gyak07
    
    samtools view -Sb illesztes01.sam > illesztes01.bam

.. code:: bash

    %%bash
    
    cd gyak07
    
    samtools sort illesztes01.bam > illesztes01_sorted.bam
    samtools index illesztes01_sorted.bam

.. code:: r

    # TERMINÁLBAN !!!
    
    cd gyak07
    
    samtools tview illesztes01_sorted.bam
    
    samtools tview illesztes01_sorted.bam ebola1976.fa


-  ?: súgó ablakot nyit
-  .: forward illesztésű, a referenciával megegyező nukleotid
-  ,: reverse illesztésű, a referenciával megegyező nukleotid
-  ACGT: forward illesztésű, referenciától eltérő nukleotid
-  acgt: reverse illesztésű, referenciától eltérő nukleotid
-  \*: törölt bázis

.. code:: bash

    %%bash
    
    cd gyak07
    
    samtools view -Sb -F 4 illesztes01.sam > illesztes01_van.bam 
    
    samtools view -Sb -f 4 illesztes01.sam > illesztes01_nincs.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak07
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa mem Ebola SRR1660259.fastq | samtools view -Sb -F 4 > illesztes02_van.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak07
    
    export PATH=$PATH:/home/bioinfo/bwa
    
    bwa mem Ebola SRR1660259.fastq | samtools view -Sb -F 4 | samtools sort > illesztes03_van_sorted.bam
    
    samtools index illesztes03_van_sorted.bam
    
    ls -lh

.. code:: bash

    %%bash
    
    cd gyak07
    
    # samtools mpileup illesztes03_van_sorted.bam > illesztes03_van_sorted.pileup
    
    samtools mpileup -f ebola1976.fa illesztes03_van_sorted.bam > illesztes03_van_sorted.pileup
    
    tail illesztes03_van_sorted.pileup

Oszlopok jelentése: 1. a referencia szekvencia azonosítója 2. a
szekvencia nukleotidjának pozíciója (1-ről induló sorszám) 3. a
referencia nukleotid az adott pozíción 4. az adott pozíciót lefedő
readek száma 5. az illesztett readek nukleotidja az adott pozícióban 6.
az illesztett readek adott pozícióbeli nukleotidjának minősége

További információ: https://en.wikipedia.org/wiki/Pileup\_format

.. code:: r

    # R
    library(Rsamtools)
    
    setwd('gyak07')
    
    bam.fajlom = 'illesztes03_van_sorted.bam'
    
    countBam(bam.fajlom)


.. parsed-literal::

    Loading required package: GenomeInfoDb
    Loading required package: BiocGenerics
    Loading required package: parallel
    
    Attaching package: ‘BiocGenerics’
    
    The following objects are masked from ‘package:parallel’:
    
        clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
        clusterExport, clusterMap, parApply, parCapply, parLapply,
        parLapplyLB, parRapply, parSapply, parSapplyLB
    
    The following objects are masked from ‘package:stats’:
    
        IQR, mad, sd, var, xtabs
    
    The following objects are masked from ‘package:base’:
    
        anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,
        colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
        grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,
        mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
        rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
        table, tapply, union, unique, unsplit, which, which.max, which.min
    
    Loading required package: S4Vectors
    Loading required package: stats4
    
    Attaching package: ‘S4Vectors’
    
    The following object is masked from ‘package:base’:
    
        expand.grid
    
    Loading required package: IRanges
    Loading required package: GenomicRanges
    Loading required package: Biostrings
    Loading required package: XVector
    
    Attaching package: ‘Biostrings’
    
    The following object is masked from ‘package:base’:
    
        strsplit
    


::


    Error in setwd("gyak07"): cannot change working directory
    Traceback:


    1. setwd("gyak07")


.. code:: r

    # R
    idxstatsBam(bam.fajlom)

.. code:: r

    # R
    # átlagos lefedettség
    
    ref.szek.hossza = idxstatsBam(bam.fajlom)$seqlength
    illesztett.nukleotidok.szama = countBam(bam.fajlom)$nucleotides
    
    illesztett.nukleotidok.szama / ref.szek.hossza

.. code:: r

    # R
    # pileup
    
    olvasasi.parameterek = PileupParam(max_depth=50)
    
    my.pileup = pileup(bam.fajlom, pileupParam=olvasasi.parameterek)
    
    head(my.pileup)


.. code:: r

    # R
    # http://ggplot2.org/
    
    library(ggplot2)
    
    ggplot(data=my.pileup, aes(x=pos, y=count)) + geom_point()


.. code:: r

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count)) + geom_point() + theme_bw()

.. code:: r

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count, color=strand)) + geom_point() + theme_bw()

.. code:: r

    # R
    ggplot(data=my.pileup, aes(x=pos, y=count, color=strand)) + geom_point() + theme_bw() + geom_smooth()

.. code:: r

    # R
    library(Gviz)
    
    # https://www.bioconductor.org/packages/release/bioc/html/Gviz.html
    # https://www.bioconductor.org/packages/release/bioc/vignettes/Gviz/inst/doc/Gviz.pdf
    
    options(ucscChromosomeNames=FALSE)
    # https://www.ncbi.nlm.nih.gov/nuccore/KM034562.1
    # http://hgdownload.cse.ucsc.edu/downloads.html#ebola_virus
    
    illesztes.track = AlignmentsTrack(bam.fajlom, start=2000, end=3000)
    
    plotTracks(
      list(illesztes.track), 
      type=c('coverage', 'pileup'), 
      chromosome='AF086833.2', 
      from=2000, 
      to=3000
    )

.. code:: r

    # R
    tengely.track = GenomeAxisTrack()
    
    plotTracks(
      list(tengely.track, illesztes.track), 
      type=c('coverage', 'pileup'), 
      chromosome='AF086833.2', 
      from=2000, 
      to=3000
    )

.. code:: r

    # R
    library(rentrez)
    library(seqinr)
    
    ebola = entrez_fetch(db='nuccore', id='AF086833.2', rettype='fasta')
    referencia = read.fasta(textConnection(ebola), as.string=TRUE, seqonly=TRUE)
    referencia.szekvencia = referencia[[1]]
    referencia.szekvencia = DNAStringSet(referencia.szekvencia)
    names(referencia.szekvencia) = 'AF086833.2'
    
    szekvencia.track = SequenceTrack(referencia.szekvencia)
    
    plotTracks(
      trackList = list(tengely.track, illesztes.track, szekvencia.track), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2000, 
      to = 3000
    )

.. code:: r

    # R
    kiemeles = HighlightTrack(
      trackList = list(illesztes.track, szekvencia.track),
      chromosome = 'AF086833.2',
      start = 2218,
      end = 2222
    )
    
    plotTracks(
      list(tengely.track, kiemeles), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2200, 
      to = 2240
    )

.. code:: r

    # R
    displayPars(kiemeles)$fill = 'blue'
    displayPars(kiemeles)$col = 'transparent'
    displayPars(kiemeles)$alpha = 0.3
                
    plotTracks(
      list(tengely.track, kiemeles), 
      type = c('coverage', 'pileup'), 
      chromosome = 'AF086833.2', 
      from = 2200, 
      to = 2240
    )
